<!--?xml version="1.0" encoding="utf-8"?-->
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
        <!-- AppResources meta begin -->
        <script type="text/javascript">var ncbi_startTime = new Date();</script>
        <!-- AppResources meta end -->
        
        <!-- TemplateResources meta begin -->
        <meta name="paf_template" content="">

        <!-- TemplateResources meta end -->
        
        <!-- Logger begin -->
        <meta name="ncbi_db" content="pmc"><meta name="ncbi_pdid" content="article"><meta name="ncbi_acc" content=""><meta name="ncbi_domain" content="ejbiosysbio"><meta name="ncbi_report" content="record"><meta name="ncbi_type" content="fulltext"><meta name="ncbi_objectid" content=""><meta name="ncbi_pcid" content="/articles/PMC4867776/"><meta name="ncbi_app" content="pmc">
        <!-- Logger end -->
        
        <title>The prediction of virus mutation using neural networks and rough set techniques</title>
        
        <!-- AppResources external_resources begin -->
        <link rel="stylesheet" href="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/jig.css"><script type="text/javascript" src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/jig.js"></script>

        <!-- AppResources external_resources end -->
        
        <!-- Page meta begin -->
        <meta name="robots" content="INDEX,NOFOLLOW,NOARCHIVE"><link rel="canonical" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/"><link rel="schema.DC" href="http://purl.org/DC/elements/1.0/"><meta name="citation_journal_title" content="EURASIP Journal on Bioinformatics and Systems Biology"><meta name="citation_title" content="The prediction of virus mutation using neural networks and rough set techniques"><meta name="citation_authors" content="Mostafa A. Salama, Aboul Ella Hassanien, Ahmad Mostafa"><meta name="citation_date" content="December 2016"><meta name="citation_volume" content="2016"><meta name="citation_doi" content="10.1186/s13637-016-0042-0"><meta name="citation_abstract_html_url" content="/pmc/articles/PMC4867776/?report=abstract"><meta name="citation_pmid" content="27257410"><meta name="DC.Title" content="The prediction of virus mutation using neural networks and rough set techniques"><meta name="DC.Type" content="Text"><meta name="DC.Publisher" content="Springer"><meta name="DC.Contributor" content="Mostafa A. Salama"><meta name="DC.Contributor" content="Aboul Ella Hassanien"><meta name="DC.Contributor" content="Ahmad Mostafa"><meta name="DC.Date" content="2016 Dec"><meta name="DC.Identifier" content="10.1186/s13637-016-0042-0"><meta name="DC.Language" content="en"><meta property="og:title" content="The prediction of virus mutation using neural networks and rough set techniques"><meta property="og:type" content="article"><meta property="og:description" content="Viral evolution remains to be a main obstacle in the effectiveness of antiviral treatments. The ability to predict this evolution will help in the early detection of drug-resistant strains and will potentially facilitate the design of more efficient antiviral ..."><meta property="og:url" content="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/"><meta property="og:site_name" content="PubMed Central (PMC)"><meta property="og:image" content="https://www.ncbi.nlm.nih.gov/corehtml/pmc/pmcgifs/pmc-logo-share.png"><meta name="twitter:card" content="summary"><meta name="twitter:site" content="@ncbi"><link rel="stylesheet" href="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/pmc.css" type="text/css"><link rel="stylesheet" href="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/pmc_extras_prnt.css" type="text/css" media="print"><script type="text/javascript" src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/common.js">//</script><script type="text/javascript" src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/NcbiTagServer.js">//</script><meta name="citationexporter" content="backend:'/pmc/utils/ctxp/'"><script type="text/javascript" src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/jquery.js">//</script><link rel="stylesheet" href="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/citationexporter.css" type="text/css"><script type="text/javascript" src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/MathJax.js"></script><script type="text/javascript" src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/large-obj-scrollbars.js"></script><script type="text/javascript">window.name="mainwindow";</script><style type="text/css">.pmc-wm {background:transparent repeat-y top left;background-image:url(/corehtml/pmc/pmcgifs/wm-ejbiosysbio.gif);background-size: auto, contain}</style><style type="text/css">.print-view{display:block}</style><style type="text/css">
        div.pmc_para_cit li.highlight,
        div.pmc_para_cit li.highlight .one_line_source
        { background: #E0E0E0; }
        a.bibr.highlight { background: #E0E0E0; } 
      </style><meta name="cited_in_systematic_reviews" content=""><link rel="alternate" type="application/epub+zip" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/epub/"><link rel="alternate" type="application/pdf" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/pdf/13637_2016_Article_42.pdf">

        <!-- Page meta end -->
    <link rel="shortcut icon" href="https://www.ncbi.nlm.nih.gov/favicon.ico"><meta name="ncbi_phid" content="F4FC0CDA94C8AC010000000000A2008E">
<meta name="referrer" content="origin-when-cross-origin"><link type="text/css" rel="stylesheet" href="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/4143404.css"><link type="text/css" rel="stylesheet" href="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/4157116.css" media="print"><style type="text/css">.MathJax_Preview {color: #888}
#MathJax_Message {position: fixed; left: 1px; bottom: 2px; background-color: #E6E6E6; border: 1px solid #959595; margin: 0px; padding: 2px 8px; z-index: 102; color: black; font-size: 80%; width: auto; white-space: nowrap}
#MathJax_MSIE_Frame {position: absolute; top: 0; left: 0; width: 0px; z-index: 101; border: 0px; margin: 0px; padding: 0px}
.MathJax_Error {color: #CC0000; font-style: italic}
</style><style type="text/css" media="print">.print-log { position:absolute;left:-10000px;top:auto;width:1px;height:1px;overflow:hidden; }.print-log li { list-style-image: url('https://www.ncbi.nlm.nih.gov/stat?jsevent=print&ncbi_app=pmc&ncbi_db=pmc&ncbi_pcid=%2Farticles%2FPMC4867776%2F&ncbi_pdid=article&ncbi_phid=F4FC0CDA94C8AC010000000000A2008E'); }</style><style type="text/css">.print-log { position:absolute;left:-10000px;top:auto;width:1px;height:1px;overflow:hidden; }</style><style type="text/css">#MathJax_About {position: fixed; left: 50%; width: auto; text-align: center; border: 3px outset; padding: 1em 2em; background-color: #DDDDDD; color: black; cursor: default; font-family: message-box; font-size: 120%; font-style: normal; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; z-index: 201; border-radius: 15px; -webkit-border-radius: 15px; -moz-border-radius: 15px; -khtml-border-radius: 15px; box-shadow: 0px 10px 20px #808080; -webkit-box-shadow: 0px 10px 20px #808080; -moz-box-shadow: 0px 10px 20px #808080; -khtml-box-shadow: 0px 10px 20px #808080; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
#MathJax_About.MathJax_MousePost {outline: none}
.MathJax_Menu {position: absolute; background-color: white; color: black; width: auto; padding: 2px; border: 1px solid #CCCCCC; margin: 0; cursor: default; font: menu; text-align: left; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; z-index: 201; box-shadow: 0px 10px 20px #808080; -webkit-box-shadow: 0px 10px 20px #808080; -moz-box-shadow: 0px 10px 20px #808080; -khtml-box-shadow: 0px 10px 20px #808080; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
.MathJax_MenuItem {padding: 2px 2em; background: transparent}
.MathJax_MenuArrow {position: absolute; right: .5em; padding-top: .25em; color: #666666; font-size: .75em}
.MathJax_MenuActive .MathJax_MenuArrow {color: white}
.MathJax_MenuArrow.RTL {left: .5em; right: auto}
.MathJax_MenuCheck {position: absolute; left: .7em}
.MathJax_MenuCheck.RTL {right: .7em; left: auto}
.MathJax_MenuRadioCheck {position: absolute; left: 1em}
.MathJax_MenuRadioCheck.RTL {right: 1em; left: auto}
.MathJax_MenuLabel {padding: 2px 2em 4px 1.33em; font-style: italic}
.MathJax_MenuRule {border-top: 1px solid #CCCCCC; margin: 4px 1px 0px}
.MathJax_MenuDisabled {color: GrayText}
.MathJax_MenuActive {background-color: Highlight; color: HighlightText}
.MathJax_MenuDisabled:focus, .MathJax_MenuLabel:focus {background-color: #E8E8E8}
.MathJax_ContextMenu:focus {outline: none}
.MathJax_ContextMenu .MathJax_MenuItem:focus {outline: none}
#MathJax_AboutClose {top: .2em; right: .2em}
.MathJax_Menu .MathJax_MenuClose {top: -10px; left: -10px}
.MathJax_MenuClose {position: absolute; cursor: pointer; display: inline-block; border: 2px solid #AAA; border-radius: 18px; -webkit-border-radius: 18px; -moz-border-radius: 18px; -khtml-border-radius: 18px; font-family: 'Courier New',Courier; font-size: 24px; color: #F0F0F0}
.MathJax_MenuClose span {display: block; background-color: #AAA; border: 1.5px solid; border-radius: 18px; -webkit-border-radius: 18px; -moz-border-radius: 18px; -khtml-border-radius: 18px; line-height: 0; padding: 8px 0 6px}
.MathJax_MenuClose:hover {color: white!important; border: 2px solid #CCC!important}
.MathJax_MenuClose:hover span {background-color: #CCC!important}
.MathJax_MenuClose:hover:focus {outline: none}
</style><style type="text/css">.MathJax_Hover_Frame {border-radius: .25em; -webkit-border-radius: .25em; -moz-border-radius: .25em; -khtml-border-radius: .25em; box-shadow: 0px 0px 15px #83A; -webkit-box-shadow: 0px 0px 15px #83A; -moz-box-shadow: 0px 0px 15px #83A; -khtml-box-shadow: 0px 0px 15px #83A; border: 1px solid #A6D ! important; display: inline-block; position: absolute}
.MathJax_Menu_Button .MathJax_Hover_Arrow {position: absolute; cursor: pointer; display: inline-block; border: 2px solid #AAA; border-radius: 4px; -webkit-border-radius: 4px; -moz-border-radius: 4px; -khtml-border-radius: 4px; font-family: 'Courier New',Courier; font-size: 9px; color: #F0F0F0}
.MathJax_Menu_Button .MathJax_Hover_Arrow span {display: block; background-color: #AAA; border: 1px solid; border-radius: 3px; line-height: 0; padding: 4px}
.MathJax_Hover_Arrow:hover {color: white!important; border: 2px solid #CCC!important}
.MathJax_Hover_Arrow:hover span {background-color: #CCC!important}
</style><style type="text/css">#MathJax_Zoom {position: absolute; background-color: #F0F0F0; overflow: auto; display: block; z-index: 301; padding: .5em; border: 1px solid black; margin: 0; font-weight: normal; font-style: normal; text-align: left; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; -webkit-box-sizing: content-box; -moz-box-sizing: content-box; box-sizing: content-box; box-shadow: 5px 5px 15px #AAAAAA; -webkit-box-shadow: 5px 5px 15px #AAAAAA; -moz-box-shadow: 5px 5px 15px #AAAAAA; -khtml-box-shadow: 5px 5px 15px #AAAAAA; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
#MathJax_ZoomOverlay {position: absolute; left: 0; top: 0; z-index: 300; display: inline-block; width: 100%; height: 100%; border: 0; padding: 0; margin: 0; background-color: white; opacity: 0; filter: alpha(opacity=0)}
#MathJax_ZoomFrame {position: relative; display: inline-block; height: 0; width: 0}
#MathJax_ZoomEventTrap {position: absolute; left: 0; top: 0; z-index: 302; display: inline-block; border: 0; padding: 0; margin: 0; background-color: white; opacity: 0; filter: alpha(opacity=0)}
</style><style type="text/css">.MathJax_Display {text-align: center; margin: 1em 0em; position: relative; display: block!important; text-indent: 0; max-width: none; max-height: none; min-width: 0; min-height: 0; width: 100%}
.MathJax .merror {background-color: #FFFF88; color: #CC0000; border: 1px solid #CC0000; padding: 1px 3px; font-style: normal; font-size: 90%}
.MathJax .MJX-monospace {font-family: monospace}
.MathJax .MJX-sans-serif {font-family: sans-serif}
#MathJax_Tooltip {background-color: InfoBackground; color: InfoText; border: 1px solid black; box-shadow: 2px 2px 5px #AAAAAA; -webkit-box-shadow: 2px 2px 5px #AAAAAA; -moz-box-shadow: 2px 2px 5px #AAAAAA; -khtml-box-shadow: 2px 2px 5px #AAAAAA; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true'); padding: 3px 4px; z-index: 401; position: absolute; left: 0; top: 0; width: auto; height: auto; display: none}
.MathJax {display: inline; font-style: normal; font-weight: normal; line-height: normal; font-size: 100%; font-size-adjust: none; text-indent: 0; text-align: left; text-transform: none; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; direction: ltr; max-width: none; max-height: none; min-width: 0; min-height: 0; border: 0; padding: 0; margin: 0}
.MathJax:focus, body :focus .MathJax {display: inline-table}
.MathJax img, .MathJax nobr, .MathJax a {border: 0; padding: 0; margin: 0; max-width: none; max-height: none; min-width: 0; min-height: 0; vertical-align: 0; line-height: normal; text-decoration: none}
img.MathJax_strut {border: 0!important; padding: 0!important; margin: 0!important; vertical-align: 0!important}
.MathJax span {display: inline; position: static; border: 0; padding: 0; margin: 0; vertical-align: 0; line-height: normal; text-decoration: none}
.MathJax nobr {white-space: nowrap!important}
.MathJax img {display: inline!important; float: none!important}
.MathJax * {transition: none; -webkit-transition: none; -moz-transition: none; -ms-transition: none; -o-transition: none}
.MathJax_Processing {visibility: hidden; position: fixed; width: 0; height: 0; overflow: hidden}
.MathJax_Processed {display: none!important}
.MathJax_ExBox {display: block!important; overflow: hidden; width: 1px; height: 60ex; min-height: 0; max-height: none}
.MathJax .MathJax_EmBox {display: block!important; overflow: hidden; width: 1px; height: 60em; min-height: 0; max-height: none}
.MathJax .MathJax_HitBox {cursor: text; background: white; opacity: 0; filter: alpha(opacity=0)}
.MathJax .MathJax_HitBox * {filter: none; opacity: 1; background: transparent}
#MathJax_Tooltip * {filter: none; opacity: 1; background: transparent}
@font-face {font-family: MathJax_Main; src: url('https://www.ncbi.nlm.nih.gov/core/mathjax/2.6.1/fonts/HTML-CSS/TeX/woff/MathJax_Main-Regular.woff?rev=2.6.1') format('woff'), url('https://www.ncbi.nlm.nih.gov/core/mathjax/2.6.1/fonts/HTML-CSS/TeX/otf/MathJax_Main-Regular.otf?rev=2.6.1') format('opentype')}
@font-face {font-family: MathJax_Main; src: url('https://www.ncbi.nlm.nih.gov/core/mathjax/2.6.1/fonts/HTML-CSS/TeX/woff/MathJax_Main-Bold.woff?rev=2.6.1') format('woff'), url('https://www.ncbi.nlm.nih.gov/core/mathjax/2.6.1/fonts/HTML-CSS/TeX/otf/MathJax_Main-Bold.otf?rev=2.6.1') format('opentype'); font-weight: bold}
@font-face {font-family: MathJax_Main; src: url('https://www.ncbi.nlm.nih.gov/core/mathjax/2.6.1/fonts/HTML-CSS/TeX/woff/MathJax_Main-Italic.woff?rev=2.6.1') format('woff'), url('https://www.ncbi.nlm.nih.gov/core/mathjax/2.6.1/fonts/HTML-CSS/TeX/otf/MathJax_Main-Italic.otf?rev=2.6.1') format('opentype'); font-style: italic}
@font-face {font-family: MathJax_Math; src: url('https://www.ncbi.nlm.nih.gov/core/mathjax/2.6.1/fonts/HTML-CSS/TeX/woff/MathJax_Math-Italic.woff?rev=2.6.1') format('woff'), url('https://www.ncbi.nlm.nih.gov/core/mathjax/2.6.1/fonts/HTML-CSS/TeX/otf/MathJax_Math-Italic.otf?rev=2.6.1') format('opentype'); font-style: italic}
@font-face {font-family: MathJax_Caligraphic; src: url('https://www.ncbi.nlm.nih.gov/core/mathjax/2.6.1/fonts/HTML-CSS/TeX/woff/MathJax_Caligraphic-Regular.woff?rev=2.6.1') format('woff'), url('https://www.ncbi.nlm.nih.gov/core/mathjax/2.6.1/fonts/HTML-CSS/TeX/otf/MathJax_Caligraphic-Regular.otf?rev=2.6.1') format('opentype')}
@font-face {font-family: MathJax_Size1; src: url('https://www.ncbi.nlm.nih.gov/core/mathjax/2.6.1/fonts/HTML-CSS/TeX/woff/MathJax_Size1-Regular.woff?rev=2.6.1') format('woff'), url('https://www.ncbi.nlm.nih.gov/core/mathjax/2.6.1/fonts/HTML-CSS/TeX/otf/MathJax_Size1-Regular.otf?rev=2.6.1') format('opentype')}
@font-face {font-family: MathJax_Size2; src: url('https://www.ncbi.nlm.nih.gov/core/mathjax/2.6.1/fonts/HTML-CSS/TeX/woff/MathJax_Size2-Regular.woff?rev=2.6.1') format('woff'), url('https://www.ncbi.nlm.nih.gov/core/mathjax/2.6.1/fonts/HTML-CSS/TeX/otf/MathJax_Size2-Regular.otf?rev=2.6.1') format('opentype')}
@font-face {font-family: MathJax_Size3; src: url('https://www.ncbi.nlm.nih.gov/core/mathjax/2.6.1/fonts/HTML-CSS/TeX/woff/MathJax_Size3-Regular.woff?rev=2.6.1') format('woff'), url('https://www.ncbi.nlm.nih.gov/core/mathjax/2.6.1/fonts/HTML-CSS/TeX/otf/MathJax_Size3-Regular.otf?rev=2.6.1') format('opentype')}
@font-face {font-family: MathJax_Size4; src: url('https://www.ncbi.nlm.nih.gov/core/mathjax/2.6.1/fonts/HTML-CSS/TeX/woff/MathJax_Size4-Regular.woff?rev=2.6.1') format('woff'), url('https://www.ncbi.nlm.nih.gov/core/mathjax/2.6.1/fonts/HTML-CSS/TeX/otf/MathJax_Size4-Regular.otf?rev=2.6.1') format('opentype')}
</style></head>
    <body class="article" id="ui-ncbiexternallink-3"><div style="visibility: hidden; overflow: hidden; position: absolute; top: 0px; height: 1px; width: auto; padding: 0px; border: 0px none; margin: 0px; text-align: left; text-indent: 0px; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal;"><div id="MathJax_Hidden"></div></div><div id="MathJax_Message" style="display: none;"></div>
        <div class="grid">
            <div class="col twelve_col nomargin shadow">
                <!-- System messages like service outage or JS required; this is handled by the TemplateResources portlet -->
                <div class="sysmessages">
                    <noscript>
	<p class="nojs">
	<strong>Warning:</strong>
	The NCBI web site requires JavaScript to function. 
	<a href="/guide/browsers/#enablejs" title="Learn how to enable JavaScript" target="_blank">more...</a>
	</p>
	</noscript>
                </div>
                <!--/.sysmessage-->
                <div class="wrap">
                    <div class="page">
                        <div class="top">
                            <div class="universal_header" id="universal_header"><ul class="inline_list jig-ncbimenu ui-ncbimenu resources_list ui-ncbibasicmenu orientation_hort" id="navcontent" role="menubar"><li class="ui-ncbimenu-item-leaf ui-ncbimenu-item-first ui-helper-reset ui-ncbimenu-item-no-hlt" role="presentation"><a class="ui-ncbimenu-link-first" href="https://www.ncbi.nlm.nih.gov/" role="menuitem" title="NCBI Home" id="ncbihome" accesskey="1"><span class="offscreen_noflow">NCBI</span><img src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/28977.gif" class="ncbi_logo" title="NCBI" alt="NCBI Logo"></a></li><li class="offscreen_noflow ui-ncbimenu-item-skip access"><a href="#maincontent" title="Skip to the content" tabindex="0" accesskey="3">Skip to main
                        content</a></li><li class="offscreen_noflow ui-ncbimenu-item-skip access"><a href="#navcontent" title="Skip to the navigation" tabindex="0" accesskey="4">Skip to
                        navigation</a></li><li id="resource-menu" class="topmenu ui-helper-reset ui-ncbimenu-item-first ui-helper-reset" role="presentation"><a class="ui-ncbimenu-first-link-has-submenu ui-ncbimenu-link-first topanchor expandDown" href="https://www.ncbi.nlm.nih.gov/static/header_footer_ajax/submenu/#resources" tabindex="-1" role="menuitem">Resources</a></li><li id="all-howtos-menu" class="topmenu ui-helper-reset ui-ncbimenu-item-first" role="presentation"><a class="ui-ncbimenu-first-link-has-submenu ui-ncbimenu-link-first topanchor expandDown" href="https://www.ncbi.nlm.nih.gov/static/header_footer_ajax/submenu/#howto" tabindex="-1" role="menuitem">How To</a></li><li class="offscreen_noflow ui-ncbimenu-item-skip access"><a href="https://www.ncbi.nlm.nih.gov/guide/browsers/#accesskeys" title="About My NCBI Accesskeys" tabindex="0" accesskey="0">About NCBI Accesskeys</a></li></ul><div class="myncbi"><span id="myncbiusername" style="display:none"><a href="https://www.ncbi.nlm.nih.gov/account/settings/" id="mnu" title="Edit account settings"></a></span><a accesskey="2" href="https://www.ncbi.nlm.nih.gov/myncbi/" id="myncbi" style="display:none">My NCBI</a><a href="https://www.ncbi.nlm.nih.gov/account/?back_url=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC4867776%2F" id="sign_in">Sign in to NCBI</a><a href="https://www.ncbi.nlm.nih.gov/account/signout/?back_url=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC4867776%2F" id="sign_out" style="display:none">Sign Out</a></div></div>
                            <div class="header">
    <div class="res_logo">
  <h1 class="img_logo"><a href="https://www.ncbi.nlm.nih.gov/pmc/" class="pmc_logo offscreen">PMC</a></h1>
  <div class="NLMLogo">
    <a href="https://www.nlm.nih.gov/" title="US National Library of Medicine">US National Library of Medicine</a>
    <br>
    <a href="https://www.nih.gov/" title="National Institutes of Health">National Institutes of Health</a>
  </div>
</div>
    <div class="search"><form method="get" action="/pmc/"><div class="search_form"><label for="database" class="offscreen_noflow">Search database</label><select id="database"><optgroup label="Recent"><option value="pmc" selected="selected" class="last" data-ac_dict="pmc-search-autocomplete">PMC</option></optgroup><optgroup label="All"><option value="gquery">All Databases</option><option value="assembly">Assembly</option><option value="biocollections">Biocollections</option><option value="bioproject">BioProject</option><option value="biosample">BioSample</option><option value="biosystems">BioSystems</option><option value="books">Books</option><option value="clinvar">ClinVar</option><option value="clone">Clone</option><option value="cdd">Conserved Domains</option><option value="gap">dbGaP</option><option value="dbvar">dbVar</option><option value="nucest">EST</option><option value="gene">Gene</option><option value="genome">Genome</option><option value="gds">GEO DataSets</option><option value="geoprofiles">GEO Profiles</option><option value="nucgss">GSS</option><option value="gtr">GTR</option><option value="homologene">HomoloGene</option><option value="medgen">MedGen</option><option value="mesh">MeSH</option><option value="ncbisearch">NCBI Web Site</option><option value="nlmcatalog">NLM Catalog</option><option value="nuccore">Nucleotide</option><option value="omim">OMIM</option><option value="pmc" data-ac_dict="pmc-search-autocomplete">PMC</option><option value="popset">PopSet</option><option value="probe">Probe</option><option value="protein">Protein</option><option value="proteinclusters">Protein Clusters</option><option value="pcassay">PubChem BioAssay</option><option value="pccompound">PubChem Compound</option><option value="pcsubstance">PubChem Substance</option><option value="pubmed">PubMed</option><option value="pubmedhealth">PubMed Health</option><option value="snp">SNP</option><option value="sparcle">Sparcle</option><option value="sra">SRA</option><option value="structure">Structure</option><option value="taxonomy">Taxonomy</option><option value="toolkit">ToolKit</option><option value="toolkitall">ToolKitAll</option><option value="toolkitbook">ToolKitBook</option><option value="toolkitbookgh">ToolKitBookgh</option><option value="unigene">UniGene</option></optgroup></select><div class="nowrap"><label for="term" class="offscreen_noflow" accesskey="/">Search term</label><div class="nowrap"><div class="jig-ncbiclearbutton-wrap ui-ncbiclearbutton-wrap"><input name="term" id="term" title="Search PMC. Use up and down arrows to choose an item from the autocomplete." class="jig-ncbiclearbutton jig-ncbiautocomplete" data-jigconfig="dictionary:'pmc-search-autocomplete',disableUrl:'NcbiSearchBarAutoComplCtrl'" autocomplete="off" data-sbconfig="ds:'no',pjs:'no',afs:'yes'" aria-haspopup="true" aria-autocomplete="list" role="textbox" type="text"><a class="reset" href="#" style="visibility: hidden;"><img src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/clear.png" alt="Clear input"></a></div></div><button id="search" type="submit" class="button_search nowrap" cmd="go">Search</button></div></div></form><ul class="searchlinks inline_list"><li>
                        <a href="https://www.ncbi.nlm.nih.gov/pmc/advanced/">Advanced</a>
                    </li><li>
                        <a href="https://www.ncbi.nlm.nih.gov/pmc/journals/">Journal list</a>
                    </li><li class="help">
                        <a target="_blank" href="https://www.ncbi.nlm.nih.gov/books/NBK3825/">Help</a>
                    </li></ul></div>
</div>

                            
                            
                        <!--<component id="Page" label="headcontent"/>-->
                            
                        </div>
                        <div class="content">
                            <!-- site messages -->
                            <div class="container">
    <div id="maincontent" class="content eight_col col">
        <div class="navlink-box">
            <ul class="page-breadcrumbs inline_list small"><li class="journal-list"><a href="https://www.ncbi.nlm.nih.gov/pmc/journals/" class="navlink">Journal List</a></li><li class="archive"><a class="navlink" href="https://www.ncbi.nlm.nih.gov/pmc/journals/495/">EURASIP J Bioinform Syst Biol</a></li><li class="issue-page"><a class="navlink" href="https://www.ncbi.nlm.nih.gov/pmc/issues/285916/">v.2016; 2016 Dec</a></li><li class="accid">PMC4867776</li></ul>
        </div>

        <!-- Journal banner -->
        <div class="pmc-page-banner whole_rhythm"><div><img src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/logo-ejbiosysbio.gif" alt="Logo of ejbiosysbio" usemap="#logo-imagemap"><map id="logo-imagemap" name="logo-imagemap"><area shape="rect" coords="0,0,500,75" alt="EURASIP Journal on Bioinformatics and Systems Biology" title="EURASIP Journal on Bioinformatics and Systems Biology" href="http://bsb.eurasipjournals.springeropen.com/" ref="reftype=publisher&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CBanner&amp;TO=Publisher%7COther%7CN/A&amp;rendering-type=normal" target="pmc_ext"></map></div> </div>
        
        <!--component id='MainPortlet' label='search-reference'/-->
        
        <!-- Book content -->
        <div class="">
            
        
            
            <div class="hide-overflow article lit-style content pmc-wm slang-all page-box"><!--main-content--><div class="jig-ncbiinpagenav" data-jigconfig="smoothScroll: false, allHeadingLevels: ['h2'], headingExclude: ':hidden'" id="ui-ncbiinpagenav-1"><div class="fm-sec half_rhythm no_top_margin"><div class="fm-citation half_rhythm no_top_margin clearfix"><div class="small"><div class="inline_block eight_col va_top"><div><div><span class="cit"><span role="menubar"><a href="#" role="menuitem" aria-expanded="false" aria-haspopup="true">EURASIP J Bioinform Syst Biol</a></span>. 2016 Dec; 2016: 10. </span></div><div><span class="fm-vol-iss-date">Published online 2016 May 13. </span>  <span class="doi">doi:&nbsp; <a href="https://dx.doi.org/10.1186%2Fs13637-016-0042-0" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CFront%20Matter&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">10.1186/s13637-016-0042-0</a></span></div></div></div><div class="inline_block four_col va_top show-overflow align_right"><div class="fm-citation-ids"><div class="fm-citation-pmcid"><span class="fm-citation-ids-label">PMCID: </span><span>PMC4867776</span></div></div></div></div></div><h1 class="content-title">The prediction of virus mutation using neural networks and rough set techniques</h1><div class="half_rhythm"><div class="contrib-group fm-author"><a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=Salama%20MA%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=27257410">Mostafa A. Salama</a>,<sup><img src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/corrauth.gif" alt="corresponding author"></sup><sup>1,</sup><sup>3</sup> <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=Hassanien%20AE%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=27257410">Aboul Ella Hassanien</a>,<sup>2,</sup><sup>3</sup> and  <a href="https://www.ncbi.nlm.nih.gov/pubmed/?term=Mostafa%20A%5BAuthor%5D&amp;cauthor=true&amp;cauthor_uid=27257410">Ahmad Mostafa</a><sup>1</sup></div></div><div class="fm-panel small half_rhythm"><div class="fm-authors-info fm-panel hide half_rhythm" id="idm140672725519168_ai" style="display:none"><div class="fm-affl" lang="en"><sup>1</sup>British University in Egypt (BUE), Cairo, Egypt </div><div class="fm-affl" lang="en"><sup>2</sup>Cairo University, Cairo, Egypt </div><div class="fm-affl" lang="en"><sup>3</sup>Scientific Research Group in Egypt, (SRGE), Cairo, Egypt </div><div><span class="fm-affl">Mostafa A. Salama, </span><span class="fm-affl"><span class="email-label">Email: </span><a href="mailto:dev@null" data-email="ge.ude.eub@amalas.afatsom" class="oemail">ge.ude.eub@amalas.afatsom</a></span>.</div><div><sup><img src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/corrauth.gif" alt="corresponding author"></sup>Corresponding author.</div></div><div class="togglers"><a href="#" class="pmctoggle" rid="idm140672725519168_ai">Author information <span>►</span></a> <a href="#" class="pmctoggle" rid="idm140672725519168_an">Article notes <span>►</span></a> <a href="#" class="pmctoggle" rid="idm140672725519168_cpl">Copyright and License information <span>►</span></a></div><div class="fm-article-notes fm-panel hide half_rhythm" id="idm140672725519168_an" style="display:none"><div class="fm-pubdate half_rhythm">Received 2015 Jul 13; Accepted 2016 May 3.</div></div><div class="fm-cpl-info fm-panel hide half_rhythm" id="idm140672725519168_cpl" style="display:none"><div class="fm-copyright half_rhythm"><a href="https://www.ncbi.nlm.nih.gov/pmc/about/copyright/">Copyright</a> © Salama et al. 2016</div><div class="fm-copyright half_rhythm">
<strong>Open Access</strong> This article is distributed under the terms of the Creative Commons Attribution 4.0 International License (<a href="http://creativecommons.org/licenses/by/4.0/" ref="reftype=extlink&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CFront%20Matter&amp;TO=External%7CLink%7CURI&amp;rendering-type=normal" target="pmc_ext">http://creativecommons.org/licenses/by/4.0/</a>),
 which permits unrestricted use, distribution, and reproduction in any 
medium, provided you give appropriate credit to the original author(s) 
and the source, provide a link to the Creative Commons license, and 
indicate if changes were made.</div></div></div><div id="pmclinksbox" class="links-box whole_rhythm hidden"></div></div><div class="sec"></div><div id="Abs1" class="tsec sec" lang="en"><div class="goto jig-ncbiinpagenav-goto-container"><span role="menubar"><a class="tgt_dark page-toc-label jig-ncbiinpagenav-goto-heading" href="#" title="Go to other sections in this page" role="menuitem" aria-expanded="false" aria-haspopup="true">Go to:</a></span></div><h2 class="head no_bottom_margin ui-helper-clearfix" id="Abs1title">Abstract</h2><!--article-meta--><div><p id="__p1" class="p p-first-last">Viral
 evolution remains to be a main obstacle in the effectiveness of 
antiviral treatments. The ability to predict this evolution will help in
 the early detection of drug-resistant strains and will potentially 
facilitate the design of more efficient antiviral treatments. Various 
tools has been utilized in genome studies to achieve this goal. One of 
these tools is machine learning, which facilitates the study of 
structure-activity relationships, secondary and tertiary structure 
evolution prediction, and sequence error correction. This work proposes a
 novel machine learning technique for the prediction of the possible 
point mutations that appear on alignments of primary RNA sequence 
structure. It predicts the genotype of each nucleotide in the RNA 
sequence, and proves that a nucleotide in an RNA sequence changes based 
on the other nucleotides in the sequence. Neural networks technique is 
utilized in order to predict new strains, then a rough set theory based 
algorithm is introduced to extract these point mutation patterns. This 
algorithm is applied on a number of aligned RNA isolates time-series 
species of the Newcastle virus. Two different data sets from two sources
 are used in the validation of these techniques. The results show that 
the accuracy of this technique in predicting the nucleotides in the new 
generation is as high as 75 %. The mutation rules are visualized for the
 analysis of the correlation between different nucleotides in the same 
RNA sequence.</p></div><div class="sec"><strong class="kwd-title">Keywords: </strong><span class="kwd-text">Gene prediction, RNA, Machine learning</span></div></div><div id="Sec1" class="tsec sec"><div class="goto jig-ncbiinpagenav-goto-container"><a class="tgt_dark page-toc-label jig-ncbiinpagenav-goto-heading" href="#" title="Go to other sections in this page" role="button" aria-expanded="false" aria-haspopup="true">Go to:</a></div><h2 class="head no_bottom_margin ui-helper-clearfix" id="Sec1title">Introduction</h2><p id="__p2" class="p p-first">The
 deoxyribonucleic acid (DNA) strands are composed of units of 
nucleotides. Each nucleotide is composed of a nitrogen-containing 
nucleobase, which is either guanine (G), adenine (A), thymine (T), or 
cytosine (C). Most DNA molecules consist of two strands coiled around 
each other forming a double helix. These DNA strands are used as a 
template to create the ribonucleic acid (RNA) in a process known as 
transcription. However, unlike DNA, RNA is often found as a 
single-strand. One type of RNA is the messenger RNA (mRNA) which carries
 information from the ribosome, which are where the protein is 
synthesized. The sequence of mRNA is what specifies the sequence of 
amino acids the formed protein. DNA and RNA are also the main component 
of viruses. Some of the viruses are DNA-based, while others are 
RNA-based such as Newcastle, HIV, and flu [<a href="#CR1" rid="CR1" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905769" role="button" aria-expanded="false" aria-haspopup="true">1</a>].
 RNA viruses are different than the DNA-based viruses in the sense that 
they have higher mutation rates, and hence, they have higher adaptive 
capacity. This mutation causes a continuous evolution that leads to host
 immunity, and hence, the virus becomes even more virulent [<a href="#CR2" rid="CR2" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905760" role="button" aria-expanded="false" aria-haspopup="true">2</a>].
 One of the RNA virus mutations is the point mutation which is a small 
scale mutations that affects the RNA sequence in only one or few 
nucleotides, such as nucleotide substitution. This substitution refers 
to the replacement of one nucleobase (i.e., base) either through 
transition or transversion. This substitution refers to the replacement 
of one nucleobase (i.e., base) by another either through transition or 
transversion. Transition is the exchange of two purines (A &lt;−&gt; G) 
or one pyrimidine (C &lt;−&gt; T), while transversion is the exchange 
between a purine and pyrimidine [<a href="#CR3" rid="CR3" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905752" role="button" aria-expanded="false" aria-haspopup="true">3</a>, <a href="#CR4" rid="CR4" class=" bibr popnode" role="button" aria-expanded="false" aria-haspopup="true">4</a>]. Another type of point mutation is frame-shift which refers to the insertion or deletion of a nucleotide in the RNA sequence.</p><p id="__p3">One of the important focuses in the field of human disease genetics is the prediction of genetic mutation [<a href="#CR5" rid="CR5" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905763" role="button" aria-expanded="false" aria-haspopup="true">5</a>].
 Having information about the current virus generations and their past 
evolution could provide a general understanding of the dynamics of virus
 evolution and the prediction of future viruses and diseases [<a href="#CR6" rid="CR6" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905765" role="button" aria-expanded="false" aria-haspopup="true">6</a>].
 The evolutionary relationship between species is determined by 
phylogenetic analysis; additionally, it infers the ancestor sequence of 
these species. These phylogenetic relationships among RNA sequences can 
help in predicting which sequence might have an equivalent function [<a href="#CR7" rid="CR7" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905757" role="button" aria-expanded="false" aria-haspopup="true">7</a>].
 The analysis of the mutation data is very important, and one of the 
tools used for this purpose is machine learning. Machine learning 
techniques help predict the effects of non-synonymous single nucleotide 
polymorphisms on protein stability, function and drug resistance [<a href="#CR8" rid="CR8" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_588022073" role="button" aria-expanded="false" aria-haspopup="true">8</a>].
 Some of these techniques that are used in prediction are support vector
 machine, neural networks and decision trees. These techniques have been
 utilized to learn the rules describing mutations that affect protein 
behavior, and use them to infer new relevant mutations that will be 
resistant to certain drugs [<a href="#CR9" rid="CR9" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905762" role="button" aria-expanded="false" aria-haspopup="true">9</a>]. Another use is to predict the potential secondary structure formation based on primary structure sequences [<a href="#CR10" rid="CR10" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905779" role="button" aria-expanded="false" aria-haspopup="true">10</a>–<a href="#CR12" rid="CR12" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905759" role="button" aria-expanded="false" aria-haspopup="true">12</a>].
 A different direction is to predict the discovery of single nucleotide 
variants in RNA sequence. Another tool in machine learning is Markov 
chains, which can describe the relative rates of different nucleotide 
changes in the RNA sequence [<a href="#CR13" rid="CR13" class=" bibr popnode" role="button" aria-expanded="false" aria-haspopup="true">13</a>].
 These models consider the RNA sequence to be a string of four discrete 
states, and hence, tracks the nucleotide replacements during the 
evolution of the sequence. In these models, it was assumed that the 
different nucleotides in the sequence evolved independently and 
identically, and justified that using the case of neutral evolution of 
nucleotides. Following that, several researches developed methods 
negated that assumption and identified the relevant neighbor-dependent 
substitution processes [<a href="#CR14" rid="CR14" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_588022071" role="button" aria-expanded="false" aria-haspopup="true">14</a>].</p><p id="__p4">The
 prediction of the mutated RNA gives a clear understanding of the 
mutation process, the future activity of the RNA, and the help direct 
the development of the drugs that should be designed for it [<a href="#CR15" rid="CR15" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905754" role="button" aria-expanded="false" aria-haspopup="true">15</a>, <a href="#CR16" rid="CR16" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905775" role="button" aria-expanded="false" aria-haspopup="true">16</a>].
 In this work, we propose a machine learning technique that is based on 
rough set theory. This technique predicts the potential nucleotide 
substitutions that may occur in primary RNA sequences. In this 
technique, a training phase is utilized in which each iteration the 
input is an RNA sequence of one generation of the virus, while the 
output is the RNA sequence of the next generation of the virus. Every 
feature in the input is a nucleotide in the RNA sequence corresponds to a
 feature in the output. The training of the machine learning technique 
is fed with aligned RNA sequences of successive generations of the same 
RNA viruses that exhibited similar environmental conditions. The 
technique introduced in this paper predicts the last RNA sequence based 
on the previous RNA generation sequence. Following that, the predicted 
RNA sequence is compared the actual RNA sequence in order to validate 
the ability of the machine learning technique to predict the RNA 
evolution and this prediction accuracy. This comparison results in a 
percentage which is calculated based on the number of matched 
nucleotides between both the predicted and actual RNA sequence versus 
the total number of nucleotides in the sequence. One of the main 
important methods of this technique is that it extracts the rules 
governing the past mutations, and hence, is able to infer the possible 
future mutations. These rules show the effect of a set of nucleotides on
 the mutation of their neighboring nucleotide. This technique visualizes
 these rules to allow an integrated analysis of the mutations occurring 
in successive generations of the RNA virus. Besides using the Rough set 
theory in our technique, a traditional machine learning technique is 
utilized which is neural networks in order to clarify the prediction 
process and validate our results. Finally, we present a way to reform 
the RNA alignments of a set of successive generations of the same virus.
 This reformation step is important in order to fit the input 
requirements for any machine learning technique.</p><p id="__p5" class="p p-last">This
 paper presents a proof of concept by applying this technique on a set 
of successive generations of the Newcastle Disease Virus (NDV) from two 
different countries, Korea and China [<a href="#CR17" rid="CR17" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905753" role="button" aria-expanded="false" aria-haspopup="true">17</a>, <a href="#CR18" rid="CR18" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905764" role="button" aria-expanded="false" aria-haspopup="true">18</a>].
 In these sets, the percentage of nucleotides that exhibits variation 
from a generation to another is 57–65 % for the two used sets of 
sequences. The proposed techniques percentage accuracy in the prediction
 of the varied nucleotides in the tested sequence in the last generation
 are 68–76 %. Although these results are not statistically significant 
at this point, it still however proves the applicability of the proposed
 technique. It is worth noting that the learning and accuracy of 
prediction of this technique increases as the number of instances in the
 data set increases. The rest of the paper is organized as follows. 
Section <a href="#Sec2" rid="Sec2" class=" sec">2</a> presents the 
related work in applying machine learning techniques in genetic 
problems, as well as the proposed technique to solve these problems. The
 experimental work and discussion appear in Sections <a href="#Sec7" rid="Sec7" class=" sec">3</a> and <a href="#Sec10" rid="Sec10" class=" sec">4</a>, and finally, the conclusion is presented in Section <a href="#Sec12" rid="Sec12" class=" sec">5</a>.</p></div><div id="Sec2" class="tsec sec"><div class="goto jig-ncbiinpagenav-goto-container"><a class="tgt_dark page-toc-label jig-ncbiinpagenav-goto-heading" href="#" title="Go to other sections in this page" role="button" aria-expanded="false" aria-haspopup="true">Go to:</a></div><h2 class="head no_bottom_margin ui-helper-clearfix" id="Sec2title">Methods</h2><div id="Sec3" class="sec sec-first"><h3>Related work</h3><p id="__p6" class="p p-first">Predicting
 techniques have been utilized in the field of genetics for many years, 
and have been geared towards different directions. One of these 
directions is the detection of the resistance of the virus to drugs 
after its mutation [<a href="#CR19" rid="CR19" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_588022072" role="button" aria-expanded="false" aria-haspopup="true">19</a>],
 in which, machine learning techniques are focused on learning the rules
 characterizing the aligned sequences that are resistant to drugs. The 
rules will be later on used to detect the drug resistance gene sequences
 amongst a set of testing sequences. The training phase of these 
techniques works by having each protein sequence represented as a 
feature vector and fed as the input to this technique [<a href="#CR20" rid="CR20" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905755" role="button" aria-expanded="false" aria-haspopup="true">20</a>].
 An example of these machine learning techniques are support vector 
machine (SVM) and neural networks, which can be trained on data sets of 
drug-resistant and non-resistant DNA sequences of virus population. In 
the training phase, these techniques learn the rules of classifying new 
generations of the virus a being drug-resistant or not [<a href="#CR21" rid="CR21" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905756" role="button" aria-expanded="false" aria-haspopup="true">21</a>].
 However, these techniques are black box techniques, that cannot be used
 to infer any information about the rules used in this classification. 
Another disadvantage is that these techniques utilize 20 bit binary 
numbers instead of characters as a representation for 20 different amino
 acids condons. This increases the size of the input to the used 
algorithm, which in return increases the complexity of the 
classification process. However, the authors in [<a href="#CR22" rid="CR22" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_588022074" role="button" aria-expanded="false" aria-haspopup="true">22</a>]
 introduced a technique that uses characters instead of number as the 
input, and they utilize a decision tree in order to provide direct 
evidence on the drug resistance genes through a set of rules. Their 
technique trains and tests its efficiency on 471 isolates against 14 
antiviral drugs, creating a decision tree for each drug. The input of 
this decision tree is the isolates sequence in which, each position is 
one of 20 naturally occurring amino acids represented in characters. The
 results of this technique concludes whether the virus is a 
drug-resgstenc virus or not. An example of this generated decision tree 
is: if the codon at position 184 is for methionine (M) and the codon at 
position 75 for alanine (A), glutamic acid (E) or threonine (T), then 
the virus carrying this gene is resistant to (3TC) antivirus drug. 
Another example is the detection of whether a point mutation in a cell 
is transmitted to the offspring or not, i.e., gremlin mutation vs. 
somatic mutation [<a href="#CR23" rid="CR23" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905770" role="button" aria-expanded="false" aria-haspopup="true">23</a>].</p><p id="__p7">Another
 research direction is the prediction of the secondary structure of the 
RNA/DNA of the organism in the generation post-mutation. The ribonucleic
 acid (RNA) molecule consists of a sequence of ribo-nucleotides that 
determines the amino acids’ sequence in the protein. The primary 
structure of the RNA molecule is the linear form of the nucleotides’ 
sequence. The nucleotides can be paired based on specific rules that is:
 adenine (A) pairs with uracil (U) and cytosine (C) with guanine (G). 
Base pairs can occur within single-stranded nucleic acids. The RNA 
sequence is folded into secondary structure in which a pair of basis is 
bonded together. This structure contains a set of canonical base pairs, 
whose variation is considered as a form of mutation that can be 
predicted. Several researches have been focused on automating the RNA 
sequence folding [<a href="#CR24" rid="CR24" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905766" role="button" aria-expanded="false" aria-haspopup="true">24</a>] and predicting the secondary structure form [<a href="#CR25" rid="CR25" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905777" role="button" aria-expanded="false" aria-haspopup="true">25</a>].
 The probability of the generation of any secondary structure is 
inversely proportional to the energy of this structure. This energy is 
modeled based on extensive thermodynamic measurements [<a href="#CR26" rid="CR26" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905751" role="button" aria-expanded="false" aria-haspopup="true">26</a>].
 Applications like “RNAMute” analyzes the effects of point mutations on 
RNAs secondary structure based on thermodynamic parameters [<a href="#CR27" rid="CR27" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905774" role="button" aria-expanded="false" aria-haspopup="true">27</a>].</p><p id="__p8">The
 third research direction is the prediction of single nucleotide 
variants (SNV) at each locus, i.e., nucleotide location. The SNV 
existence are identified from the results of the Next Generation 
Sequencing (NGS) methods [<a href="#CR6" rid="CR6" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905773" role="button" aria-expanded="false" aria-haspopup="true">6</a>].
 NSG is capable of typing SNP, which is the mutation that produces a 
single base pair change in the DNA sequence. For example, the two 
sequenced DNA fragments from different generations are AAGCCTA and 
AAGCTTA. Noticed that the fifth single nucleotide C in the first 
fragment varied to a T in the second fragment. This genotype variation 
change represents the mutation in the child genomes of the next 
generation. The steps of allocating the SNVs start with collecting a set
 of aligned sequences from NGS readings analyzed against a reference 
sequence. At each position <em>i</em> in the genome data, the number of reads <em>a</em>
<sub><em>i</em></sub> that match the reference genome and the number of reads <em>b</em>
<sub><em>i</em></sub> that do not match the reference genome are counted. The total number of reads, depth, is given by <em>N</em>
<sub><em>i</em></sub>=<em>a</em>
<sub><em>i</em></sub> + <em>b</em>
<sub><em>i</em></sub>. A naïve approach to detect the SNV locations is to find the location <em>i</em> ∈ [1, T] whose fraction <em>f</em>
<sub><em>i</em></sub>=<em>a</em>
<sub><em>i</em></sub> / <em>N</em>
<sub><em>i</em></sub> is less than a certain threshold [<a href="#CR28" rid="CR28" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905776" role="button" aria-expanded="false" aria-haspopup="true">28</a>].</p><p id="__p9">Although
 this approach is accurate for large number of generations, usually this
 is not case due to low collected number of sequences. Moreover, it only
 predicts the existence of the SNV, however, it does not predict the 
future sequence. A model is proposed in [<a href="#CR29" rid="CR29" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905778" role="button" aria-expanded="false" aria-haspopup="true">29</a>]
 to infer the genotype at each location. The model characterizes each 
column in the alignment to be one of three states, the first Homozygous 
type is where all genotypes are the same as the reference allele states,
 while the second types is where all genotypes are the same as 
non-reference allele, the last type is for mixture of reference and 
non-reference alleles. This model calculates posterior probability P(<em>g</em> | <em>x</em>,<em>z</em>) of the genotype at position <em>u</em> in the current sequence <em>z</em>, given a reference sequence <em>x</em>
 and the sequence z. The genotype highest posterior probability (MAP) is
 selected. The detection of the third state is based on SNVMix model [<a href="#CR30" rid="CR30" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905761" role="button" aria-expanded="false" aria-haspopup="true">30</a>], which uses the Bayesian Theorem and MAP to calculate the posterior probabilities for a mixed genotype <em>g</em>
<sub><em>m</em></sub>. In this case, the <em>P</em>(<em>g</em>
<sub><em>m</em></sub>|<em>a</em>
<sub><em>i</em></sub>) can be calculated as shown in Eq. <a href="#Equ1" rid="Equ1" class=" ">1</a> where <em>a</em>
<sub><em>i</em></sub> is the number of reads that matches the reference at location <em>i</em>. 
</p><div class="disp-formula" id="Equ1"><div class="f"><em>P</em>(<em>g</em><sub><em>m</em></sub>|<em>a</em><sub><em>i</em></sub>) =&nbsp;<em>P</em>(<em>a</em><sub><em>i</em></sub>|<em>g</em><sub><em>m</em></sub>) ∗ <em>P</em>(<em>g</em>)</div><div class="l">1</div></div><p>
</p><p id="__p10">The prior probability <em>P</em>(<em>a</em>
<sub><em>i</em></sub>|<em>g</em>
<sub><em>m</em></sub>) is calculated using the binomial distribution <em>dbinom</em> in the case of mixed genotype [<a href="#CR30" rid="CR30" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905772" role="button" aria-expanded="false" aria-haspopup="true">30</a>] as shown in Eqs. <a href="#Equ2" rid="Equ2" class=" ">2</a>, <a href="#Equ3" rid="Equ3" class=" ">3</a>, and <a href="#Equ4" rid="Equ4" class=" ">4</a>. The parameter <span id="IEq1"><em>μ</em><sub><em>g</em><sub><em>m</em></sub></sub></span> of the <em>dbinom</em> is of value 0.5 where the probability of the genotype, matching or not matching, is equal. And <em>N</em>
<sub><em>i</em></sub> represents the total number of generations. 
</p><div class="disp-formula" id="Equ2"><div class="f"><em>P</em>(<em>a</em><sub><em>i</em></sub>|<em>g</em><sub><em>m</em></sub>)&nbsp;≈&nbsp;dbinom(<em>a</em><sub><em>i</em></sub>|<em>μ</em><sub><em>g</em><sub><em>m</em></sub></sub>,&nbsp;<em>N</em><sub><em>i</em></sub>)</div><div class="l">2</div></div><p>
</p><p id="__p11">
</p><div class="disp-formula" id="Equ3"><div class="f"><span class="MathJax_Preview"></span><div class="MathJax_Display" style="text-align: center;"><span class="MathJax" id="MathJax-Element-1-Frame" tabindex="0" style=""><nobr><span class="math" id="M8" overflow="scroll" role="math" style="width: 19.266em; display: inline-block;"><span style="display: inline-block; position: relative; width: 15.524em; height: 0px; font-size: 124%;"><span style="position: absolute; clip: rect(1.827em, 1015.34em, 4.528em, -1000em); top: -3.427em; left: 0em;"><span class="mrow" id="MathJax-Span-2"><span class="mi" id="MathJax-Span-3" style="font-family: MathJax_Math; font-style: italic;">P<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.109em;"></span></span><span class="mo" id="MathJax-Span-4" style="vertical-align: 0em;"><span style="font-family: MathJax_Size3;">(</span></span><span class="msub" id="MathJax-Span-5"><span style="display: inline-block; position: relative; width: 0.848em; height: 0px;"><span style="position: absolute; clip: rect(3.39em, 1000.51em, 4.143em, -1000em); top: -3.982em; left: 0em;"><span class="mrow" id="MathJax-Span-6"><span class="mi" id="MathJax-Span-7" style="font-family: MathJax_Math; font-style: italic;">a</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; top: -3.832em; left: 0.529em;"><span class="mrow" id="MathJax-Span-8"><span class="mi" id="MathJax-Span-9" style="font-size: 70.7%; font-family: MathJax_Math; font-style: italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span><span class="mo" id="MathJax-Span-10" style="vertical-align: 1.449em;"><span style="display: inline-block; position: relative; width: 0.278em; height: 0px;"><span style="position: absolute; font-family: MathJax_Main; top: -3.232em; left: 0em;">∣<span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; font-family: MathJax_Main; top: -1.832em; left: 0em;">∣<span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="font-family: MathJax_Main; position: absolute; top: -2.532em; left: 0em;">∣<span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span><span class="msub" id="MathJax-Span-11"><span style="display: inline-block; position: relative; width: 1.173em; height: 0px;"><span style="position: absolute; clip: rect(3.389em, 1000.48em, 4.338em, -1000em); top: -3.982em; left: 0em;"><span class="mrow" id="MathJax-Span-12"><span class="mi" id="MathJax-Span-13" style="font-family: MathJax_Math; font-style: italic;">g<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.003em;"></span></span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; top: -3.741em; left: 0.477em;"><span class="mrow" id="MathJax-Span-14"><span class="mi" id="MathJax-Span-15" style="font-size: 70.7%; font-family: MathJax_Math; font-style: italic;">m</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span><span class="mo" id="MathJax-Span-16" style="vertical-align: 0em;"><span style="font-family: MathJax_Size3;">)</span></span><span class="mo" id="MathJax-Span-17" style="font-family: MathJax_Main; padding-left: 0.278em;">=</span><span class="mfenced" id="MathJax-Span-18" style="padding-left: 0.278em;"><span class="mo" id="MathJax-Span-77" style="vertical-align: 0em;"><span style="font-family: MathJax_Size3;">(</span></span><span class="mfrac" id="MathJax-Span-20"><span style="display: inline-block; position: relative; width: 1.122em; height: 0px; margin-right: 0.12em; margin-left: 0.12em;"><span style="position: absolute; clip: rect(3.148em, 1001.12em, 4.291em, -1000em); top: -4.658em; left: 50%; margin-left: -0.561em;"><span class="mrow" id="MathJax-Span-21"><span class="msub" id="MathJax-Span-22"><span style="display: inline-block; position: relative; width: 1.122em; height: 0px;"><span style="position: absolute; clip: rect(3.148em, 1000.89em, 4.133em, -1000em); top: -3.982em; left: 0em;"><span class="mrow" id="MathJax-Span-23"><span class="mi" id="MathJax-Span-24" style="font-family: MathJax_Math; font-style: italic;">N<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.085em;"></span></span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; top: -3.832em; left: 0.803em;"><span class="mrow" id="MathJax-Span-25"><span class="mi" id="MathJax-Span-26" style="font-size: 70.7%; font-family: MathJax_Math; font-style: italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; clip: rect(3.39em, 1000.85em, 4.291em, -1000em); top: -3.296em; left: 50%; margin-left: -0.424em;"><span class="mrow" id="MathJax-Span-27"><span class="msub" id="MathJax-Span-28"><span style="display: inline-block; position: relative; width: 0.848em; height: 0px;"><span style="position: absolute; clip: rect(3.39em, 1000.51em, 4.143em, -1000em); top: -3.982em; left: 0em;"><span class="mrow" id="MathJax-Span-29"><span class="mi" id="MathJax-Span-30" style="font-family: MathJax_Math; font-style: italic;">a</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; top: -3.832em; left: 0.529em;"><span class="mrow" id="MathJax-Span-31"><span class="mi" id="MathJax-Span-32" style="font-size: 70.7%; font-family: MathJax_Math; font-style: italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span><span class="mo" id="MathJax-Span-78" style="vertical-align: 0em;"><span style="font-family: MathJax_Size3;">)</span></span></span><span class="msubsup" id="MathJax-Span-36"><span style="display: inline-block; position: relative; width: 1.507em; height: 0px;"><span style="position: absolute; clip: rect(3.389em, 1000.58em, 4.349em, -1000em); top: -3.982em; left: 0em;"><span class="mrow" id="MathJax-Span-37"><span class="mi" id="MathJax-Span-38" style="font-family: MathJax_Math; font-style: italic;">μ</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; clip: rect(3.519em, 1000.67em, 4.245em, -1000em); top: -4.438em; left: 0.603em;"><span class="mrow" id="MathJax-Span-39"><span class="msub" id="MathJax-Span-40"><span style="display: inline-block; position: relative; width: 0.6em; height: 0px;"><span style="position: absolute; clip: rect(3.519em, 1000.36em, 4.14em, -1000em); top: -3.982em; left: 0em;"><span class="mrow" id="MathJax-Span-41"><span class="mi" id="MathJax-Span-42" style="font-size: 70.7%; font-family: MathJax_Math; font-style: italic;">a</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; top: -3.876em; left: 0.374em;"><span class="mrow" id="MathJax-Span-43"><span class="mi" id="MathJax-Span-44" style="font-size: 50%; font-family: MathJax_Math; font-style: italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; clip: rect(3.518em, 1000.9em, 4.309em, -1000em); top: -3.834em; left: 0.603em;"><span class="mrow" id="MathJax-Span-45"><span class="msub" id="MathJax-Span-46"><span style="display: inline-block; position: relative; width: 0.829em; height: 0px;"><span style="position: absolute; clip: rect(3.518em, 1000.34em, 4.278em, -1000em); top: -3.982em; left: 0em;"><span class="mrow" id="MathJax-Span-47"><span class="mi" id="MathJax-Span-48" style="font-size: 70.7%; font-family: MathJax_Math; font-style: italic;">g<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.002em;"></span></span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; top: -3.812em; left: 0.337em;"><span class="mrow" id="MathJax-Span-49"><span class="mi" id="MathJax-Span-50" style="font-size: 50%; font-family: MathJax_Math; font-style: italic;">m</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span><span class="mfenced" id="MathJax-Span-51" style="padding-left: 0.167em;"><span class="mo" id="MathJax-Span-79" style="vertical-align: 0em;"><span style="font-family: MathJax_Size2;">(</span></span><span class="mrow" id="MathJax-Span-53"><span class="mn" id="MathJax-Span-54" style="font-family: MathJax_Main;">1</span><span class="mo" id="MathJax-Span-55" style="font-family: MathJax_Main; padding-left: 0.222em;">−</span><span class="msubsup" id="MathJax-Span-56" style="padding-left: 0.222em;"><span style="display: inline-block; position: relative; width: 2.247em; height: 0px;"><span style="position: absolute; clip: rect(3.389em, 1000.58em, 4.349em, -1000em); top: -3.982em; left: 0em;"><span class="mrow" id="MathJax-Span-57"><span class="mi" id="MathJax-Span-58" style="font-family: MathJax_Math; font-style: italic;">μ</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; clip: rect(3.348em, 1001.64em, 4.245em, -1000em); top: -4.438em; left: 0.603em;"><span class="mrow" id="MathJax-Span-59"><span class="msub" id="MathJax-Span-60"><span style="display: inline-block; position: relative; width: 0.793em; height: 0px;"><span style="position: absolute; clip: rect(3.348em, 1000.63em, 4.133em, -1000em); top: -3.982em; left: 0em;"><span class="mrow" id="MathJax-Span-61"><span class="mi" id="MathJax-Span-62" style="font-size: 70.7%; font-family: MathJax_Math; font-style: italic;">N<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.06em;"></span></span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; top: -3.876em; left: 0.568em;"><span class="mrow" id="MathJax-Span-63"><span class="mi" id="MathJax-Span-64" style="font-size: 50%; font-family: MathJax_Math; font-style: italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span><span class="msub" id="MathJax-Span-65"><span style="display: inline-block; position: relative; width: 0.776em; height: 0px;"><span style="position: absolute; clip: rect(3.64em, 1000.49em, 3.97em, -1000em); top: -3.982em; left: 0em;"><span class="mrow" id="MathJax-Span-66"><span class="mo" id="MathJax-Span-67" style="font-size: 70.7%; font-family: MathJax_Main;">−</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; top: -3.876em; left: 0.55em;"><span class="mrow" id="MathJax-Span-68"><span class="mi" id="MathJax-Span-69" style="font-size: 50%; font-family: MathJax_Math; font-style: italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; clip: rect(3.518em, 1000.9em, 4.309em, -1000em); top: -3.834em; left: 0.603em;"><span class="mrow" id="MathJax-Span-70"><span class="msub" id="MathJax-Span-71"><span style="display: inline-block; position: relative; width: 0.829em; height: 0px;"><span style="position: absolute; clip: rect(3.518em, 1000.34em, 4.278em, -1000em); top: -3.982em; left: 0em;"><span class="mrow" id="MathJax-Span-72"><span class="mi" id="MathJax-Span-73" style="font-size: 70.7%; font-family: MathJax_Math; font-style: italic;">g<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.002em;"></span></span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; top: -3.812em; left: 0.337em;"><span class="mrow" id="MathJax-Span-74"><span class="mi" id="MathJax-Span-75" style="font-size: 50%; font-family: MathJax_Math; font-style: italic;">m</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span></span><span class="mo" id="MathJax-Span-80" style="vertical-align: 0em;"><span style="font-family: MathJax_Size2;">)</span></span></span></span><span style="display: inline-block; width: 0px; height: 3.427em;"></span></span></span><span style="display: inline-block; overflow: hidden; vertical-align: -1.24em; border-left: 0px solid; width: 0px; height: 3.1em;"></span></span></nobr></span></div><script type="math/mml" id="MathJax-Element-1"><math xmlns:mml="http://www.w3.org/1998/Math/MathML" display="block" id="M8" overflow="scroll"><mi>P</mi><mo>(</mo><msub><mrow><mi>a</mi></mrow><mrow><mi>i</mi></mrow></msub><mo>|</mo><msub><mrow><mi>g</mi></mrow><mrow><mi>m</mi></mrow></msub><mo>)</mo><mo>=</mo><mfenced close=")" open="(" separators=""><mfrac linethickness="0.0pt"><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mrow><msub><mrow><mi>a</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></mfrac></mfenced><msubsup><mrow><mi>μ</mi></mrow><mrow><msub><mrow><mi>g</mi></mrow><mrow><mi>m</mi></mrow></msub></mrow><mrow><msub><mrow><mi>a</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></msubsup><mfenced close=")" open="(" separators=""><mrow><mn>1</mn><mo>−</mo><msubsup><mrow><mi>μ</mi></mrow><mrow><msub><mrow><mi>g</mi></mrow><mrow><mi>m</mi></mrow></msub></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi>i</mi></mrow></msub><msub><mrow><mo>−</mo></mrow><mrow><mi>i</mi></mrow></msub></mrow></msubsup></mrow></mfenced></math></script></div><div class="l">3</div></div><p>
</p><p id="__p12">
</p><div class="disp-formula" id="Equ4"><div class="f"><span class="MathJax_Preview"></span><div class="MathJax_Display" style="text-align: center;"><span class="MathJax" id="MathJax-Element-2-Frame" tabindex="0" style=""><nobr><span class="math" id="M10" overflow="scroll" role="math" style="width: 14.203em; display: inline-block;"><span style="display: inline-block; position: relative; width: 11.442em; height: 0px; font-size: 124%;"><span style="position: absolute; clip: rect(1.827em, 1011.44em, 4.528em, -1000em); top: -3.427em; left: 0em;"><span class="mrow" id="MathJax-Span-82"><span class="mi" id="MathJax-Span-83" style="font-family: MathJax_Math; font-style: italic;">P<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.109em;"></span></span><span class="mo" id="MathJax-Span-84" style="vertical-align: 0em;"><span style="font-family: MathJax_Size3;">(</span></span><span class="msub" id="MathJax-Span-85"><span style="display: inline-block; position: relative; width: 0.848em; height: 0px;"><span style="position: absolute; clip: rect(3.39em, 1000.51em, 4.143em, -1000em); top: -3.982em; left: 0em;"><span class="mrow" id="MathJax-Span-86"><span class="mi" id="MathJax-Span-87" style="font-family: MathJax_Math; font-style: italic;">a</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; top: -3.832em; left: 0.529em;"><span class="mrow" id="MathJax-Span-88"><span class="mi" id="MathJax-Span-89" style="font-size: 70.7%; font-family: MathJax_Math; font-style: italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span><span class="mo" id="MathJax-Span-90" style="vertical-align: 1.449em;"><span style="display: inline-block; position: relative; width: 0.278em; height: 0px;"><span style="position: absolute; font-family: MathJax_Main; top: -3.232em; left: 0em;">∣<span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; font-family: MathJax_Main; top: -1.832em; left: 0em;">∣<span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="font-family: MathJax_Main; position: absolute; top: -2.532em; left: 0em;">∣<span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span><span class="msub" id="MathJax-Span-91"><span style="display: inline-block; position: relative; width: 2.18em; height: 0px;"><span style="position: absolute; clip: rect(3.389em, 1000.48em, 4.338em, -1000em); top: -3.982em; left: 0em;"><span class="mrow" id="MathJax-Span-92"><span class="mi" id="MathJax-Span-93" style="font-family: MathJax_Math; font-style: italic;">g<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.003em;"></span></span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; top: -3.741em; left: 0.477em;"><span class="mrow" id="MathJax-Span-94"><span class="mi" id="MathJax-Span-95" style="font-size: 70.7%; font-family: MathJax_Math; font-style: italic;">i</span><span class="mo" id="MathJax-Span-96" style="font-size: 70.7%; font-family: MathJax_Main;">{</span><span class="mtext" id="MathJax-Span-97" style="font-size: 70.7%; font-family: MathJax_Math; font-style: italic;">ab</span><span class="mo" id="MathJax-Span-98" style="font-size: 70.7%; font-family: MathJax_Main;">}</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span><span class="mo" id="MathJax-Span-99" style="vertical-align: 0em;"><span style="font-family: MathJax_Size3;">)</span></span><span class="mo" id="MathJax-Span-100" style="font-family: MathJax_Main; padding-left: 0.278em;">=</span><span class="mfenced" id="MathJax-Span-101" style="padding-left: 0.278em;"><span class="mo" id="MathJax-Span-102" style="vertical-align: 0em;"><span style="font-family: MathJax_Size3;">(</span></span><span class="mfrac" id="MathJax-Span-103"><span style="display: inline-block; position: relative; width: 1.122em; height: 0px; margin-right: 0.12em; margin-left: 0.12em;"><span style="position: absolute; clip: rect(3.148em, 1001.12em, 4.291em, -1000em); top: -4.658em; left: 50%; margin-left: -0.561em;"><span class="mrow" id="MathJax-Span-104"><span class="msub" id="MathJax-Span-105"><span style="display: inline-block; position: relative; width: 1.122em; height: 0px;"><span style="position: absolute; clip: rect(3.148em, 1000.89em, 4.133em, -1000em); top: -3.982em; left: 0em;"><span class="mrow" id="MathJax-Span-106"><span class="mi" id="MathJax-Span-107" style="font-family: MathJax_Math; font-style: italic;">N<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.085em;"></span></span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; top: -3.832em; left: 0.803em;"><span class="mrow" id="MathJax-Span-108"><span class="mi" id="MathJax-Span-109" style="font-size: 70.7%; font-family: MathJax_Math; font-style: italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; clip: rect(3.39em, 1000.85em, 4.291em, -1000em); top: -3.296em; left: 50%; margin-left: -0.424em;"><span class="mrow" id="MathJax-Span-110"><span class="msub" id="MathJax-Span-111"><span style="display: inline-block; position: relative; width: 0.848em; height: 0px;"><span style="position: absolute; clip: rect(3.39em, 1000.51em, 4.143em, -1000em); top: -3.982em; left: 0em;"><span class="mrow" id="MathJax-Span-112"><span class="mi" id="MathJax-Span-113" style="font-family: MathJax_Math; font-style: italic;">a</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; top: -3.832em; left: 0.529em;"><span class="mrow" id="MathJax-Span-114"><span class="mi" id="MathJax-Span-115" style="font-size: 70.7%; font-family: MathJax_Math; font-style: italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span><span class="mo" id="MathJax-Span-116" style="vertical-align: 0em;"><span style="font-family: MathJax_Size3;">)</span></span></span><span class="mfrac" id="MathJax-Span-117"><span style="display: inline-block; position: relative; width: 1.488em; height: 0px; margin-right: 0.12em; margin-left: 0.12em;"><span style="position: absolute; clip: rect(3.165em, 1000.43em, 4.133em, -1000em); top: -4.658em; left: 50%; margin-left: -0.25em;"><span class="mrow" id="MathJax-Span-118"><span class="mn" id="MathJax-Span-119" style="font-family: MathJax_Main;">1</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; clip: rect(2.955em, 1001.37em, 4.133em, -1000em); top: -3.14em; left: 50%; margin-left: -0.684em;"><span class="mrow" id="MathJax-Span-120"><span class="msup" id="MathJax-Span-121"><span style="display: inline-block; position: relative; width: 1.368em; height: 0px;"><span style="position: absolute; clip: rect(3.165em, 1000.45em, 4.133em, -1000em); top: -3.982em; left: 0em;"><span class="mrow" id="MathJax-Span-122"><span class="mn" id="MathJax-Span-123" style="font-family: MathJax_Main;">2</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; top: -4.375em; left: 0.5em;"><span class="mrow" id="MathJax-Span-124"><span class="msub" id="MathJax-Span-125"><span style="display: inline-block; position: relative; width: 0.793em; height: 0px;"><span style="position: absolute; clip: rect(3.348em, 1000.63em, 4.133em, -1000em); top: -3.982em; left: 0em;"><span class="mrow" id="MathJax-Span-126"><span class="mi" id="MathJax-Span-127" style="font-size: 70.7%; font-family: MathJax_Math; font-style: italic;">N<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.06em;"></span></span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; top: -3.876em; left: 0.568em;"><span class="mrow" id="MathJax-Span-128"><span class="mi" id="MathJax-Span-129" style="font-size: 50%; font-family: MathJax_Math; font-style: italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 3.982em;"></span></span><span style="position: absolute; clip: rect(0.844em, 1001.49em, 1.21em, -1000em); top: -1.278em; left: 0em;"><span style="display: inline-block; overflow: hidden; vertical-align: 0em; border-top: 1.3px solid; width: 1.488em; height: 0px;"></span><span style="display: inline-block; width: 0px; height: 1.058em;"></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 3.427em;"></span></span></span><span style="display: inline-block; overflow: hidden; vertical-align: -1.24em; border-left: 0px solid; width: 0px; height: 3.1em;"></span></span></nobr></span></div><script type="math/mml" id="MathJax-Element-2"><math xmlns:mml="http://www.w3.org/1998/Math/MathML" display="block" id="M10" overflow="scroll"><mi>P</mi><mo>(</mo><msub><mrow><mi>a</mi></mrow><mrow><mi>i</mi></mrow></msub><mo>|</mo><msub><mrow><mi>g</mi></mrow><mrow><mi>i</mi><mo>{</mo><mtext mathvariant="italic">ab</mtext><mo>}</mo></mrow></msub><mo>)</mo><mo>=</mo><mfenced close=")" open="(" separators=""><mfrac linethickness="0.0pt"><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mrow><msub><mrow><mi>a</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></mfrac></mfenced><mfrac><mrow><mn>1</mn></mrow><mrow><msup><mrow><mn>2</mn></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></msup></mrow></mfrac></math></script></div><div class="l">4</div></div><p>
</p><p id="__p13">These three research direction focus on predicting the
 activity of the newly evolved viruses. A different direction is the 
prediction of the rates of variation of the nucleotides in the RNA 
sequences of successive generations of the virus [<a href="#CR31" rid="CR31" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_588022075" role="button" aria-expanded="false" aria-haspopup="true">31</a>].
 In this direction, models are introduced to analyze the historical 
patterns and variation mechanism of the virus and detect the rates of 
variation of each point substitution of its RNA sequence. These models 
are characterized as either neighbor-dependent or non-dependent 
variation.</p><p id="__p14" class="p p-last">The contribution in this 
paper focuses on the prediction of the RNA sequence of newly evolved 
virus by detecting the possible point mutations. The nucleotide 
substitutions of the coding regions in this Newcastle virus occurred 
frequently [<a href="#CR32" rid="CR32" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905768" role="button" aria-expanded="false" aria-haspopup="true">32</a>].
 These substitutions and dependency between nucleotides is captured to 
form a set of rules that describes the evolution of the RNA virus.</p></div><div id="Sec4" class="sec sec-last"><h3>The proposed algorithm and discussion</h3><p id="__p15" class="p p-first">The
 focus of this paper is to analyze the changes and mutations that occur 
after each generation of the virus. This analysis is done through 
monitoring these changes and detecting their patterns. The mutation of 
the RNA viruses are characterized by high drug-resistance, however, 
these mutations can be predicted by applying machine learning techniques
 in order to extract their rules and patterns. The proposed algorithm 
here is applied on a data set of aligned RNA sequences observed over a 
period of time. This data was collected and presented in previous 
research [<a href="#CR17" rid="CR17" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905771" role="button" aria-expanded="false" aria-haspopup="true">17</a>, <a href="#CR18" rid="CR18" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905780" role="button" aria-expanded="false" aria-haspopup="true">18</a>],
 in which all animal procedures performed were reviewed, approved, and 
supervised by the Institutional Animal Care and Use Committee of Konkuk 
University. This space of potential virus mutations provides a proper 
data set required for the computational methods used in the algorithm. 
The preprocessing is started by the alignment of RNA sequence for the 
purpose of predicting the evolution of the virus based on the gathered 
sequences. The steps performed are the mining and learning the rules of 
the virus mutation, in order to predict the mutations to help creating 
new drugs. The input data set is a set of aligned DNA sequences. The RNA
 isolates are ordered from old to new according to the time of getting 
the virus and isolating the RNA. The machine learning techniques used in
 this step are neural networks, and a proposed technique that is based 
on the rough set theory. An important step in the proposed technique is 
building the decision matrix required for rule extraction, which is 
based on the idea that if the value of the attribute has no effect on 
describing the category of the object, then this attribute can be 
excluded from the rule set [<a href="#CR33" rid="CR33" class=" bibr popnode" role="button" aria-expanded="false" aria-haspopup="true">33</a>].
 An important clarification is that every nucleotide in both the neural 
networks and proposed technique act as the target class label required 
in the classification. These target nucleotides will have one of four 
different genotype values that will be predicted by the classifier.</p><p id="__p16" class="p">The
 preprocessing step includes the feature selection which is important in
 decreasing the processing cost by the removal unnecessary nucleotides. 
These nucleotides are those that do not change during the different 
generations of the virus, hence, they do not add any information in the 
classification/prediction step. This is due to the fact that not 
changing across RNA generations imply that they have no effect on the 
mutated nucleotides. Hence, their existence will have no effect, and 
will moreover deteriorate the net accuracy prediction result. Although 
this step is important as a preprocessing step in the neural network 
methods, it can be avoided in the proposed rough set based technique 
since the technique removes the non-required nucleotides internally.</p><div id="Sec5" class="sec"><p></p><h4 class="inline">Neural network technique </h4><p id="__p17" class="p p-first">Neural
 networks technique can be used to predict different mutations in RNA 
sequences. The first step of this technique is to specify the structure 
of both the input and output. The number of nodes in the input and in 
the output of the neural networks is the four times the number of 
nucleotides in the RNA sequence. Hence, each nucleotide will be scaled 
to 4 binary bits, i.e., if the nucleotide is “A”, then the corresponding
 bit will be 1. However, it is not correct to transform the letters to 
numerals, for example the nucleotide values of [A, C, G, T] to [0, 1, 2,
 3]. The reason being is that the distance between each nucleotide is 
the same, while the distance between [0–3] is not the same as that 
between [0–1]. The effect of this can be demonstrated by the following 
example: when applying neural networks to the numerically transformed 
sequence, the results show unsuccessful results. This is because the 
neural networks technique changes the values of the output based on the 
nucleotide values. Hence, the backward propagation in the case of moving
 from the value <em>T</em> to the value <em>A</em> will not be the same as moving from <em>T</em> to <em>G</em>.
 This is considered a mistake in the learning phase of the neural 
network, and it will lead to incorrect classification and prediction. 
The same case occurs when applying support vector machine and Bayesian 
belief networks. The training of the neural network will occur first by 
considering every scaled RNA segment s one training input to the neural 
network. The desired output corresponding to the input DNA segment is 
the next successive scaled RNA segment from the next generation of the 
training data set. As shown in Fig. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig1/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="Fig1" rid-ob="ob-Fig1" co-legend-rid="lgnd_Fig1"><span style="position: relative;text-decoration:none;">​<span class="figpopup-sensitive-area" style="left: -2em;">Fig.1,</span></span><span>1</span></a>,
 for every input/output sequence, the weights are continuously updated 
until accuracy exceeds 70 %. This accuracy is calculated according to 
the number of correct predicted nucleotides to the total number of 
nucleotides in the sequence. After the accuracy exceeds 70 %, the next 
input is the output DNA sequence in the current step. The last RNA 
segment is left for testing.

</p><!--fig ft0--><!--fig mode=article f1--><div class="fig iconblock ten_col whole_rhythm clearfix" id="Fig1" co-legend-rid="lgnd_Fig1"><a class="icnblk_img figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig1/" target="figure" rid-figpopup="Fig1" rid-ob="ob-Fig1"><img src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/13637_2016_42_Fig1_HTML.gif" class="small-thumb" alt="Fig. 1" title="Fig. 1" src-large="/pmc/articles/PMC4867776/bin/13637_2016_42_Fig1_HTML.jpg"></a><div class="icnblk_cntnt" id="lgnd_Fig1"><div><a class="figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig1/" target="figure" rid-figpopup="Fig1" rid-ob="ob-Fig1">Fig. 1</a></div><!--caption a4--><div><span>The learning of the neural network from the input data set</span></div></div></div><p id="__p19">The
 disadvantage in using a neural network technique or support vector 
machine is the scaling of each nucleotide in the DNA sequence to four 
input states. This scaling process increase the computational complexity
 of the technique. On the other hand, the limited number of input 
instances could negatively affect the accuracy of the prediction 
process. Finally, the extraction of the rules in this technique is not 
possible, and hence, the prediction process is not feasible.</p><p id="__p20" class="p p-last">
</p><div class="figure"><img class="fig-image" alt="An external file that holds a picture, illustration, etc.
Object name is 13637_2016_42_Figa_HTML.gif" title="An external file that holds a picture, illustration, etc.
Object name is 13637_2016_42_Figa_HTML.gif" src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/13637_2016_42_Figa_HTML.gif"></div><p>
</p></div><div id="Sec6" class="sec sec-last"><p></p><h4 class="inline">Rough set gene evolution (RSGE) proposed technique </h4><p id="__p21" class="p p-first">This
 paper proposed a new algorithm for solving the evolution prediction 
problem that is based on the input time series data set. Each RNA 
sequence is mutated, i.e., evolution passage, to the next RNA sequence 
in the data set, as these sequences are sorted from old to recent dates.
 For each iteration in this learning phase of this algorithm, the input 
is an RNA sequence and the output is the next RNA sequence in the data 
set. The learned output is not the classification of the RNA virus; 
however, it is the RNA virus after mutation. Because, techniques like 
support vector machine, neural networks, and Bayesian belief networks 
have to deal with data in a numerical form, the proposed technique deals
 with alphabetical and numerical data in the same fashion.</p><p id="__p22">The purpose of this technique is to infer the rules that governs certain value [<em>A</em>,<em>C</em>,<em>G</em>
<em>o</em>
<em>r</em>
<em>T</em>] for each nucleotide. At this point, the training applied 
will consider each nucleotide as a target class of four values. The 
input that leads to one of the values of the target class is the 
sequence before the sequence containing the current value of the target 
class. The machine learning algorithm will learn what input will produce
 which output accordingly. The rules learned from the used machine 
learning algorithm will be used to predict the generated nucleotide 
corresponding to the input. The rule will be in the form of short 
sequences of nucleotides genotype and location that govern the mutation 
of a nucleotide from certain genotype to another. The number of 
iterations will be the number of nucleotides in the RNA sequence. After 
each iteration, the required rule for the nucleotide x corresponding to 
the current iteration will be extracted.</p><p id="__p23">In each iteration, the algorithm detects the value of nucleotide <em>x</em> in the sequence in the data set, as well as the value of this same nucleotide in the next sequence <em>x</em>
<sup>′</sup>. The following step in the algorithm is the detection of 
the values of all nucleotides in the preceding sequences to the ones 
that include nucleotides <em>x</em> and <em>x</em>
<sup>′</sup>. This step is applied to detect the sequence of the 
previous nucleotide values and leads to the value of the nucleotide 
under investigation. This is illustrated in Figs. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig2/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="Fig2" rid-ob="ob-Fig2" co-legend-rid="lgnd_Fig2"><span style="position: relative;text-decoration:none;">​<span class="figpopup-sensitive-area" style="left: -2.5em;">Figs.2</span></span><span>2</span></a> and <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig3/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="Fig3" rid-ob="ob-Fig3" co-legend-rid="lgnd_Fig3"><span style="position: relative;text-decoration:none;">​<span class="figpopup-sensitive-area" style="left: -1.5em;">and3.</span></span><span>3</span></a>. In Fig. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig2/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="Fig2" rid-ob="ob-Fig2" co-legend-rid="lgnd_Fig2"><span style="position: relative;text-decoration:none;">​<span class="figpopup-sensitive-area" style="left: -2em;">Fig.2,</span></span><span>2</span></a>, the sequence [<em>C</em>
<em>G</em>
<em>G</em>
<em>G</em>
<em>A</em>
<em>T</em>] precedes nucleotide <em>x</em> at position <em>i</em> of value <em>A</em>, and the sequence [<em>C</em>
<em>T</em>
<em>G</em>
<em>A</em>
<em>A</em>
<em>C</em>] precedes nucleotide <em>x</em>
<sup>′</sup> at position <em>i</em> of value <em>A</em>. The nucleotide <em>x</em> in the position <em>i</em> corresponding to the current iteration does not change from sequence <em>j</em>+1 to sequence <em>j</em>+2. In this case, if the nucleotides in the sequence preceding to the one corresponding to the nucleotide <em>x</em> do not change, then these nucleotide values will be included in the rule of nucleotide <em>x</em>, otherwise they will be completely excluded. Hence, for Fig. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig2/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="Fig2" rid-ob="ob-Fig2" co-legend-rid="lgnd_Fig2"><span style="position: relative;text-decoration:none;">​<span class="figpopup-sensitive-area" style="left: -2em;">Fig.2,</span></span><span>2</span></a>, the extracted rule for the value <em>A</em> of the nucleotide at position <em>i</em> will be [<em>A</em>:<em>i</em>,<em>C</em>:<em>i</em>+1,<em>G</em>:<em>i</em>+3,<em>A</em>:<em>i</em>+5]. A different case is presented in Fig. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig3/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="Fig3" rid-ob="ob-Fig3" co-legend-rid="lgnd_Fig3"><span style="position: relative;text-decoration:none;">​<span class="figpopup-sensitive-area" style="left: -2em;">Fig.3,</span></span><span>3</span></a>, where the value of the nucleotide at position <em>i</em> is changed from value <em>A</em> to value <em>C</em>. In this case, if the nucleotides in the sequence preceding the one corresponding to the nucleotide <em>x</em> changes, then these nucleotide values will be included in the rule of nucleotide <em>x</em>, otherwise they will be completely excluded. Hence, for Fig. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig3/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="Fig3" rid-ob="ob-Fig3" co-legend-rid="lgnd_Fig3"><span style="position: relative;text-decoration:none;">​<span class="figpopup-sensitive-area" style="left: -2em;">Fig.3,</span></span><span>3</span></a>, the extracted rule for the value <em>C</em> of the nucleotide at position i will be [<em>A</em>:<em>i</em>,<em>T</em>:<em>i</em>+2,<em>A</em>:<em>i</em>+4,<em>C</em>:<em>i</em>+6].


</p><!--fig ft0--><!--fig mode=article f1--><div class="fig iconblock ten_col whole_rhythm clearfix" id="Fig2" co-legend-rid="lgnd_Fig2"><a class="icnblk_img figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig2/" target="figure" rid-figpopup="Fig2" rid-ob="ob-Fig2"><img src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/13637_2016_42_Fig2_HTML.gif" class="small-thumb" alt="Fig. 2" title="Fig. 2" src-large="/pmc/articles/PMC4867776/bin/13637_2016_42_Fig2_HTML.jpg"></a><div class="icnblk_cntnt" id="lgnd_Fig2"><div><a class="figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig2/" target="figure" rid-figpopup="Fig2" rid-ob="ob-Fig2">Fig. 2</a></div><!--caption a4--><div><span>Nucleotide i for iteration i in the proposed algorithm, nucleotide as position i is the same, not changed</span></div></div></div><!--fig ft0--><!--fig mode=article f1--><div class="fig iconblock ten_col whole_rhythm clearfix" id="Fig3" co-legend-rid="lgnd_Fig3"><a class="icnblk_img figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig3/" target="figure" rid-figpopup="Fig3" rid-ob="ob-Fig3"><img src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/13637_2016_42_Fig3_HTML.gif" class="small-thumb" alt="Fig. 3" title="Fig. 3" src-large="/pmc/articles/PMC4867776/bin/13637_2016_42_Fig3_HTML.jpg"></a><div class="icnblk_cntnt" id="lgnd_Fig3"><div><a class="figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig3/" target="figure" rid-figpopup="Fig3" rid-ob="ob-Fig3">Fig. 3</a></div><!--caption a4--><div><span>Nucleotide i for iteration i in the proposed algorithm, nucleotide as position i is the not the same, changed</span></div></div></div><p id="__p26" class="p p-last">The
 reason behind using this methodology in extracting the rule is that for
 each iteration in this algorithm is that two main cases are considered.
 The first case is that the value of the iterated nucleotide 
corresponding to this iteration does not change. In this case, if the 
neighbor nucleotides to this iterated nucleotide do not change, this 
will indicate that the values of the nucleotides are attached and leads 
to the value of the iterated nucleotide. If the neighbor nucleotides did
 change, then the variation of these values do not affect the iterated 
nucleotide, and hence they can simply be removed from the extracted 
rule. The second case is considered the opposite of the previous one 
where the existed nucleotides in the RNA sequence will cause the change 
of the genotype of a specific nucleotide in the following generation. In
 this case, the extracted rule of the genotype at this nucleotide 
location will include the set of neighbor nucleotides. While the 
unchanged behavior of some nucleotides means their unimportance or 
non-effect of changing the value of the of iterated nucleotide. As shown
 in Algorithm 1, if the number of sequences is <em>N</em>, and the number of nucleotides is <em>K</em>, the computational complexity of the calculations is as follows: 
</p><div class="disp-formula" id="Equ5"><div class="f">AlgorithmComplexity =&nbsp;<em>O</em>(<em>K</em> ∗ <em>N</em> ∗ (2 ∗ <em>K</em>)) =&nbsp;<em>O</em>(<em>N</em> ∗ <em>K</em><sup>2</sup>)</div><div class="l">5</div></div><p>
</p></div></div></div><div id="Sec7" class="tsec sec"><div class="goto jig-ncbiinpagenav-goto-container"><a class="tgt_dark page-toc-label jig-ncbiinpagenav-goto-heading" href="#" title="Go to other sections in this page" role="button" aria-expanded="false" aria-haspopup="true">Go to:</a></div><h2 class="head no_bottom_margin ui-helper-clearfix" id="Sec7title">Results</h2><p id="__p27" class="p p-first">The
 analysis of RNA mutations requires the gathering and preparing a set of
 aligned RNA sequences that go through different mutations over a long 
period of time. A set of time-series successive isolates of the 
Newcastle virus RNA are collected from two different countries, China 
and South Korea [<a href="#CR17" rid="CR17" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905767" role="button" aria-expanded="false" aria-haspopup="true">17</a>, <a href="#CR18" rid="CR18" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_584905758" role="button" aria-expanded="false" aria-haspopup="true">18</a>]. The GenBank accession numbers of NDVs isolates recovered from live chicken markets in Korea in year 2000 are <a id="__tag_589853194" class="tag_hotlink" href="https://www.ncbi.nlm.nih.gov/nuccore/AY630409.1" ref="/nuccore/AY630409.1"><span class="tag-json" style="display:none">{"type":"entrez-nucleotide","attrs":{"text":"AY630409.1","term_id":"50511827","term_text":"AY630409.1"}}</span>AY630409.1</a>-<a id="__tag_589853195" class="tag_hotlink" href="https://www.ncbi.nlm.nih.gov/nuccore/AY630436.1" ref="/nuccore/AY630436.1"><span class="tag-json" style="display:none">{"type":"entrez-nucleotide","attrs":{"text":"AY630436.1","term_id":"50511881","term_text":"AY630436.1"}}</span>AY630436.1</a>, and from healthy domestic ducks on farms are <a id="__tag_589853193" class="tag_hotlink" href="https://www.ncbi.nlm.nih.gov/nuccore/EU547752.1" ref="/nuccore/EU547752.1"><span class="tag-json" style="display:none">{"type":"entrez-nucleotide","attrs":{"text":"EU547752.1","term_id":"170672255","term_text":"EU547752.1"}}</span>EU547752.1</a>-<a id="__tag_589853198" class="tag_hotlink" href="https://www.ncbi.nlm.nih.gov/nuccore/EU547764.1" ref="/nuccore/EU547764.1"><span class="tag-json" style="display:none">{"type":"entrez-nucleotide","attrs":{"text":"EU547764.1","term_id":"170672279","term_text":"EU547764.1"}}</span>EU547764.1</a>.
 The total number of isolates for this data set is 22, each isolate is 
of 200 nucleotide. While the accession numbers of sequenced isolates 
extracted from chicken in China in years from 2011 to 2012 are <a id="__tag_584905750" class="tag_hotlink" href="https://www.ncbi.nlm.nih.gov/nuccore/?term=KJ184574:KJ184600[pacc]" ref="/nuccore/?term=KJ184574:KJ184600[pacc]"><span class="tag-json" style="display:none">{"type":"entrez-nucleotide-range","attrs":{"text":"KJ184574-KJ184600","start_term":"KJ184574","end_term":"KJ184600","start_term_id":"592961438","end_term_id":"592961490"}}</span>KJ184574-KJ184600</a> [<a href="#CR34" rid="CR34" class=" bibr popnode tag_hotlink tag_tooltip" id="__tag_588022070" role="button" aria-expanded="false" aria-haspopup="true">34</a>].
 The total number of isolates for this data set is 45, each isolate is 
of 240 nucleotide. Each set of isolates is listed and sorted according 
to the date of extraction for time series analysis of the evolution of 
the NDV virus. The experimental is applied only a partial F gene 
sequence of NDV from the GenBank record. The virus RNA sequences were 
monitored and collected retrospectively at regular intervals from 
similar animal type. The intervals between successive RNA sequences in 
the Chinese dataset is short relative to the Korean data set. The 
difference between both types of intervals does not affect neither the 
analysis process nor the accuracy of the prediction. The extracted data 
examined for the Korean and Chinese datasets are represented by two 
different regions of viral genome with the different lengths (200 and 
240). It is important to clarify that the two input data sets can not be
 merged because each input set of segments are aligned separately. So it
 is impossible here to apply prediction performance of the proposed 
models on the Chinese dataset using the Korean dataset for training.</p><p id="__p28">Figure <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig4/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="Fig4" rid-ob="ob-Fig4" co-legend-rid="lgnd_Fig4"><span style="position: relative;text-decoration:none;">​<span class="figpopup-sensitive-area" style="left: -3em;">Figure4</span></span><span>4</span></a>
 shows a segment of these aligned RNA sequence, in which, we select only
 the columns that contain missing values. It is clear in this figure 
that some nucleotides columns are passing through mutation along the 
time period under examination, while the other nucleotides do not 
witness any change during it. Applying any machine learning technique is
 composed of two main steps, the first is the training of the first part
 of the input data set, while the second step is using the rest of the 
data for testing. The classification accuracy percentage is the ration 
of correctly predicted nucleotide divided to the total number of 
nucleotides in the sequence.

</p><!--fig ft0--><!--fig mode=article f1--><div class="fig iconblock ten_col whole_rhythm clearfix" id="Fig4" co-legend-rid="lgnd_Fig4"><a class="icnblk_img figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig4/" target="figure" rid-figpopup="Fig4" rid-ob="ob-Fig4"><img src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/13637_2016_42_Fig4_HTML.gif" class="small-thumb" alt="Fig. 4" title="Fig. 4" src-large="/pmc/articles/PMC4867776/bin/13637_2016_42_Fig4_HTML.jpg"></a><div class="icnblk_cntnt" id="lgnd_Fig4"><div><a class="figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig4/" target="figure" rid-figpopup="Fig4" rid-ob="ob-Fig4">Fig. 4</a></div><!--caption a4--><div><span>Aligned gene sequence of nucleotides</span></div></div></div><p id="__p30" class="p">The
 Chinese data set shows 33 % of the total number of nucleotides have 
been mutated, and this percentage in the Korean is 43 %. The rate of 
genome mutations over the time period applied shows variable number 
mutations.</p><div id="Sec8" class="sec"><h3>Neural network (NN) results</h3><p id="__p31" class="p p-first-last">In
 order to apply NN to the aligned time series RNA sequences, only a part
 of the sequence will be considered. The partial training of the RNA 
sequence is applied to ensure a reasonable training execution time. 
Applying neural networks to this number of target class labels will lead
 to a very high processing complexity. The training phase contains only 
20 nucleotides, where each nucleotide will be scaled to only four input 
nodes. This will lead to a data set of 80 input nodes and 80 output 
nodes. For the Korean input data set, the in-out for the training is the
 first 20 out of 22 instances. The target 20 output instances for these 
20 input instances started from the second instance in the input data 
set and ends at the 21<sup><em>s</em><em>t</em></sup> instance. The testing will have the 21<sup><em>s</em><em>t</em></sup> RNA sequence as the input to the neural networks and the 22<sup><em>n</em><em>d</em></sup>
 RNA sequence as the target output. The results show that after 3743901 
learning cycles of back-forth propagation, the validation accuracy 
reaches 70 % percent as shown in Fig. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig5/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="Fig5" rid-ob="ob-Fig5" co-legend-rid="lgnd_Fig5"><span style="position: relative;text-decoration:none;">​<span class="figpopup-sensitive-area" style="left: -2em;">Fig.5</span></span><span>5</span></a>.

</p><!--fig ft0--><!--fig mode=article f1--><div class="fig iconblock ten_col whole_rhythm clearfix" id="Fig5" co-legend-rid="lgnd_Fig5"><a class="icnblk_img figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig5/" target="figure" rid-figpopup="Fig5" rid-ob="ob-Fig5"><img src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/13637_2016_42_Fig5_HTML.gif" class="small-thumb" alt="Fig. 5" title="Fig. 5" src-large="/pmc/articles/PMC4867776/bin/13637_2016_42_Fig5_HTML.jpg"></a><div class="icnblk_cntnt" id="lgnd_Fig5"><div><a class="figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig5/" target="figure" rid-figpopup="Fig5" rid-ob="ob-Fig5">Fig. 5</a></div><!--caption a4--><div><span>Neural network classification results</span></div></div></div></div><div id="Sec9" class="sec sec-last"><h3>Rough set gene evolution (RSGE) results</h3><p id="__p33" class="p p-first">The
 validation of the proposed rough set gene evolution (RSGE) is achieved 
through testing the total number of nucleotides. For the Korean input 
data set where the total number of nucleotides is 200, the number of 
correct nucleotide matches is 148, which is considered 74 % 
classification accuracy percentage. Also, only four nucleotides have 
shown incorrect matches, which corresponds to 2 % out of all 
nucleotides. 48 nucleotides have shown no matching results, which 
indicates that none of the four rules of each of these nucleotides is 
applied. These results show the predicted and actual target RNA sequence
 of nucleotides.</p><p id="__p34">For Korea Data Set Newcastle disease virus strain Kr-XX/YY [1982–2012]: 
</p><ul class="unordered" style="list-style-type:disc"><!--
list-behavior=unordered
prefix-word=
mark-type=disc
--><li><div>Actual : ATGGGTTCCAAATCTTCTACCAG</div></li><li><div>Predicted : ATGGGNNCCANANCTTCTACCAN</div></li></ul><p>
</p><p id="__p37" class="p p-last">For the Chinese input data set where 
the total number of nucleotides is 240, the number of correct nucleotide
 matches is 180, which corresponds to 75 % classification accuracy. 
Also, only four nucleotides, i.e., 1.6 %, have shown incorrect matches. 
And, 56 nucleotides have shown no matching results, which indicates that
 non of the four rules of each of these nucleotides is applied. The 
results for the China Data Set Newcastle disease virus isolate JS XX XX 
Ch [2011–2012] is: 
</p><ul class="unordered" style="list-style-type:disc"><!--
list-behavior=unordered
prefix-word=
mark-type=disc
--><li><div>Actual : ATGGGCTCCAAACCTTCTACCAG</div></li><li><div>Predicted : TGGGCTCCAAACNTTCTACCNG</div></li></ul><p>
</p></div></div><div id="Sec10" class="tsec sec"><div class="goto jig-ncbiinpagenav-goto-container"><a class="tgt_dark page-toc-label jig-ncbiinpagenav-goto-heading" href="#" title="Go to other sections in this page" role="button" aria-expanded="false" aria-haspopup="true">Go to:</a></div><h2 class="head no_bottom_margin ui-helper-clearfix" id="Sec10title">Discussion</h2><p id="__p40" class="p p-first">A sample of the generated rules is figured as follows: 
</p><ul class="unordered" style="list-style-type:disc"><!--
list-behavior=unordered
prefix-word=
mark-type=disc
--><li><div>For Nucleotide P131: 
<ul class="unordered" style="list-style-type:disc"><!--
list-behavior=unordered
prefix-word=
mark-type=disc
--><li><div>If Genotype is A then P47=C, P57=A</div></li><li><div>If Genotype is G then P23=A</div></li></ul>
</div></li><li><div>For Nucleotide P101: 
<ul class="unordered" style="list-style-type:disc"><!--
list-behavior=unordered
prefix-word=
mark-type=disc
--><li><div>If Genotype is C then P6=T, P118=C</div></li><li><div>If Genotype is T then P6=G, P118=A</div></li></ul>
</div></li></ul><p>
</p><p id="__p47">This sample is composed of two base nucleotides only 
in the RNA sequence, which are 131 and 101. The first rule shows that 
the nucleotide at position 131 will take the genotype value “A” in the 
following generation if the nucleotide at position 47 in the current 
generation is ’C’ and the one at position 57 is of type “A”. These rules
 show the existence of a correlation between the three nucleotides, 
i.e., those at positions [131, 47, and 57], in the first rule and [101, 6
 and 118]. The genotype values of some specific nucleotides could affect
 the alteration/mutation of the RNA sequence. The input Chinese and 
Korean data sets contain different genotypes, AA, AB, and BB. The 
nucleotides of the genotypes that have the value AA and BB can be 
excluded from the resulted rule set. These genotypes need not to be 
predicted as the values are the same over all the generations. Table <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/table/Tab1/" target="true" class="fig-table-link table figpopup" rid-figpopup="Tab1" rid-ob="ob-Tab1" co-legend-rid=""><span style="position: relative;text-decoration:none;">​<span class="figpopup-sensitive-area" style="left: -2.5em;">Table1</span></span><span>1</span></a>
 shows the extracted rule sets of China countries for the AB nucleotides
 only. On the other hand, a similar table of rules is generated for the 
Korean data set.

</p><!--table ft1--><!--table-wrap mode=article t1--><div class="table-wrap iconblock ten_col whole_rhythm clearfix" id="Tab1"><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/table/Tab1/" target="table" rid-ob="ob-Tab1" rid-figpopup="Tab1" class="table img_link icnblk_img figpopup"><img alt="Table 1" title="Table 1" class="small-thumb" src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/a.gif" src-large="/pmc/articles/PMC4867776/table/Tab1/?report=previmg"></a><div class="icnblk_cntnt"><div><a class="figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/table/Tab1/" target="table" rid-figpopup="Tab1" rid-ob="ob-Tab1">Table 1</a></div><!--caption a4--><div><span>AB genotype rules for the Chinese data set</span></div></div></div><p id="__p49" class="p">Finally,
 the correlation between the nucleotides can be visualized after 
extracting the prediction rules from the RSGE technique. This form of 
exploring the effect of nucleotides on one another can provide a better 
understanding of the mutation mechanism existing in the virus’s RNA. An 
example of this correlation visual is shown in Fig. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig6/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="Fig6" rid-ob="ob-Fig6" co-legend-rid="lgnd_Fig6"><span style="position: relative;text-decoration:none;">​<span class="figpopup-sensitive-area" style="left: -2em;">Fig.6.</span></span><span>6</span></a>. In this figure, when the genotype of the nucleotide at position <em>N</em>
<sub>37</sub> is C, the genotype of the nucleotide at position <em>N</em>
<sub>33</sub> is T. And when the genotype of the nucleotide at position <em>N</em>
<sub>33</sub> is T, the genotype of the nucleotides at positions <em>N</em>
<sub>53</sub> and <em>N</em>
<sub>68</sub> are T and G respectively. The is applied on the rules generated for the Korean data set as shown in Fig. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig7/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="Fig7" rid-ob="ob-Fig7" co-legend-rid="lgnd_Fig7"><span style="position: relative;text-decoration:none;">​<span class="figpopup-sensitive-area" style="left: -2em;">Fig.7</span></span><span>7</span></a>.


</p><!--fig ft0--><!--fig mode=article f1--><div class="fig iconblock ten_col whole_rhythm clearfix" id="Fig6" co-legend-rid="lgnd_Fig6"><a class="icnblk_img figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig6/" target="figure" rid-figpopup="Fig6" rid-ob="ob-Fig6"><img src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/13637_2016_42_Fig6_HTML.gif" class="small-thumb" alt="Fig. 6" title="Fig. 6" src-large="/pmc/articles/PMC4867776/bin/13637_2016_42_Fig6_HTML.jpg"></a><div class="icnblk_cntnt" id="lgnd_Fig6"><div><a class="figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig6/" target="figure" rid-figpopup="Fig6" rid-ob="ob-Fig6">Fig. 6</a></div><!--caption a4--><div><span>Nucleotides correlation in China data set</span></div></div></div><!--fig ft0--><!--fig mode=article f1--><div class="fig iconblock ten_col whole_rhythm clearfix" id="Fig7" co-legend-rid="lgnd_Fig7"><a class="icnblk_img figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig7/" target="figure" rid-figpopup="Fig7" rid-ob="ob-Fig7"><img src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/13637_2016_42_Fig7_HTML.gif" class="small-thumb" alt="Fig. 7" title="Fig. 7" src-large="/pmc/articles/PMC4867776/bin/13637_2016_42_Fig7_HTML.jpg"></a><div class="icnblk_cntnt" id="lgnd_Fig7"><div><a class="figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig7/" target="figure" rid-figpopup="Fig7" rid-ob="ob-Fig7">Fig. 7</a></div><!--caption a4--><div><span>Nucleotides correlation in Korean data set</span></div></div></div><div id="Sec11" class="sec sec-last"><h3>NN vs. RSGE</h3><p id="__p52" class="p p-first-last">Figure <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig8/" target="figure" class="fig-table-link fig figpopup" rid-figpopup="Fig8" rid-ob="ob-Fig8" co-legend-rid="lgnd_Fig8"><span style="position: relative;text-decoration:none;">​<span class="figpopup-sensitive-area" style="left: -3em;">Figure8</span></span><span>8</span></a>
 demonstrates a comparison between the results of using neural networks 
versus the proposed rough set gene evolution prediction techniques in 
the classification of both the Chinese and Korean data sets. The results
 show a good performance of the RSGE in comparison to NN. As the data 
set increase, the preprocessing increases, and hence, the computational 
complexity of the neural networks and increases the classification 
accuracy decreases. On the other hand, the error in the classification 
using the RSGE proposed technique is approximately 2 % in 77 % of the 
sequence. The rest 23 % could not be predicted by the technique and were
 replaced by the genotype in the previous generation.

</p><!--fig ft0--><!--fig mode=article f1--><div class="fig iconblock ten_col whole_rhythm clearfix" id="Fig8" co-legend-rid="lgnd_Fig8"><a class="icnblk_img figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig8/" target="figure" rid-figpopup="Fig8" rid-ob="ob-Fig8"><img src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/13637_2016_42_Fig8_HTML.gif" class="small-thumb" alt="Fig. 8" title="Fig. 8" src-large="/pmc/articles/PMC4867776/bin/13637_2016_42_Fig8_HTML.jpg"></a><div class="icnblk_cntnt" id="lgnd_Fig8"><div><a class="figpopup" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/figure/Fig8/" target="figure" rid-figpopup="Fig8" rid-ob="ob-Fig8">Fig. 8</a></div><!--caption a4--><div><span>Prediction accuracy for Korean and Chinese data sets</span></div></div></div></div></div><div id="Sec12" class="tsec sec"><div class="goto jig-ncbiinpagenav-goto-container"><a class="tgt_dark page-toc-label jig-ncbiinpagenav-goto-heading" href="#" title="Go to other sections in this page" role="button" aria-expanded="false" aria-haspopup="true">Go to:</a></div><h2 class="head no_bottom_margin ui-helper-clearfix" id="Sec12title">Conclusions</h2><p id="__p54" class="p p-first-last">The
 contribution of machine learning techniques in the RNA mutation was 
limited to the prediction of the activity of the virus of resulted RNA. 
This work paves the way to a new horizon where the prediction of the 
mutations, such as virus evolution, is possible. It can assist the 
designing of new drugs for possible drug-resistant strains of the virus 
before a possible outbreak. Also it can help in devising diagnostics for
 the early detection of cancer and possibly for the early 
start-of-treatment. This work studies the correlation between the 
nucleotides in RNA including the effect of each nucleotide in chaining 
the genotypes of other nucleotides. These rules of these correlations 
are explored and visualized for the prediction of the mutations that may
 appear in the following generations. The prediction rules are extracted
 by a proposed technique based on RSGE, and is trained by two data sets 
extracted from two different countries. This work proves the existence 
of a correlation between the mutation of nucleotides, and successfully 
predicts the nucleotides in the next generation in the testing parts of 
two used data sets with a success rate of 75 %. On the other hand, the 
proposed rough set (RSGE) based technique shows a better prediction 
result than the neural networks technique, and moreover, it extract the 
rules used in the prediction.</p></div><div id="__ackidm140672725254304" class="tsec sec"><div class="goto jig-ncbiinpagenav-goto-container"><a class="tgt_dark page-toc-label jig-ncbiinpagenav-goto-heading" href="#" title="Go to other sections in this page" role="button" aria-expanded="false" aria-haspopup="true">Go to:</a></div><h2 class="head no_bottom_margin ui-helper-clearfix" id="__ackidm140672725254304title">Acknowledgements</h2><div class="sec"><p id="__p59">We
 thank all the members of the group of scientific research in Egypt, 
Cairo university for their continuous support and encouragement to 
accomplish this work. Besides, we thank the faculty of computer science 
in the British university in Egypt for providing us with all the 
required support. A special thanks for Mr. Saleh Esmate for helping us 
in gathering the data set of this work.</p></div></div><div id="__fn-groupidm140672725071728" class="tsec sec"><div class="goto jig-ncbiinpagenav-goto-container"><a class="tgt_dark page-toc-label jig-ncbiinpagenav-goto-heading" href="#" title="Go to other sections in this page" role="button" aria-expanded="false" aria-haspopup="true">Go to:</a></div><h2 class="head no_bottom_margin ui-helper-clearfix" id="__fn-groupidm140672725071728title">Footnotes</h2><!--back/fn-group--><div class="fm-sec half_rhythm small"><p class="fn sec" id="__fnidm140672725071600"></p><p id="__p55" class="p p-first"><strong>Competing interests</strong></p><p id="__p56" class="p p-last">The
 authors (Mostafa A. salama, Aboul-Ellah Hassnien and Ahmad Mostafa) of 
this manuscript clarify that they have no affiliations with or 
involvement in any organization or entity with any financial interest 
(such as honoraria, educational grants, participation in speakers’ 
bureaus, membership, employment, consultancies, stock ownership, or 
other equity interest, and expert testimony or patent-licensing 
arrangements), or non-financial interest (such as personal or 
professional relationships, affiliations, knowledge or beliefs) in the 
subject matter or materials discussed in this manuscript.</p><p></p><p class="fn sec" id="__fnidm140672725255808"></p><p id="__p57" class="p p-first"><strong>Authors’ contributions</strong></p><p id="__p58" class="p p-last">All authors contributes equally in this work. All authors read and approved the final manuscript.</p><p></p></div></div><div id="Bib1" class="tsec sec"><div class="goto jig-ncbiinpagenav-goto-container"><a class="tgt_dark page-toc-label jig-ncbiinpagenav-goto-heading" href="#" title="Go to other sections in this page" role="button" aria-expanded="false" aria-haspopup="true">Go to:</a></div><h2 class="head no_bottom_margin ui-helper-clearfix" id="Bib1title">References</h2><div class="ref-list-sec sec" id="reference-list"><div class="ref-cit-blk half_rhythm" id="CR1">1. <span class="element-citation">Elena SF, Sanjuán R. Adaptive value of high mutation rates of RNA viruses: separating causes from consequences. <span><span class="ref-journal">J. Virol. </span>2005;<span class="ref-vol">79</span>(18):11555–11558. doi: 10.1128/JVI.79.18.11555-11558.2005.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1212614/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/16140732" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1128%2FJVI.79.18.11555-11558.2005" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR2">2. <span class="element-citation">Wilson
 B, Garud NR, Feder AF, Assaf ZJ, Pennings PS. The population genetics 
of drug resistance evolution in natural 2 populations of viral, 
bacterial, and eukaryotic pathogens. <span><span class="ref-journal">Mol. Ecol. </span>2016;<span class="ref-vol">25</span>:42–66. doi: 10.1111/mec.13474.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4943078/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/26578204" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1111%2Fmec.13474" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR3">3. <span class="element-citation">Baranovich
 T, Wong S, Armstrong J, Marjuki H, Webby R, Webster R, Govorkova E. 
T-705 (Favipiravir) induces lethal mutagenesis in influenza A H1N1 
Viruses In Vitro. <span><span class="ref-journal">J. Virol. </span>2013;<span class="ref-vol">87</span>(7):3741–3751. doi: 10.1128/JVI.02346-12.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3624194/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/23325689" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1128%2FJVI.02346-12" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR4">4. <span class="element-citation">Loewe L. Genetic mutation. <span><span class="ref-journal">Nat. Educ. </span>2008;<span class="ref-vol">1</span>(1):113.</span></span></div><div class="ref-cit-blk half_rhythm" id="CR5">5. <span class="element-citation">Stranger BE, Dermitzakis ET. From DNA to RNA to disease and back: the ‘central dogma’ of regulatory disease variation. <span><span class="ref-journal">Hum. Genomics. </span>2006;<span class="ref-vol">2</span>(6):383–390. doi: 10.1186/1479-7364-2-6-383.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3525162/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/16848976" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1186%2F1479-7364-2-6-383" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR6">6. <span class="element-citation">Shendure J, Ji H. Next-generation DNA sequencing. <span><span class="ref-journal">Nat. Biotechnol. </span>2008;<span class="ref-vol">26</span>:1135–1145. doi: 10.1038/nbt1486.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/18846087" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1038%2Fnbt1486" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR7">7. <span class="element-citation">Xu
 J, Guo HC, Wei YQ, Shu L, Wang J, Li JS, Cao SZ, Sun SQ. Phylogenetic 
analysis of canine parvovirus isolates from Sichuan and Gansu provinces 
of China in 2011. <span><span class="ref-journal">Transbound. Emerg. Dis. </span>2015;<span class="ref-vol">62</span>:91–95. doi: 10.1111/tbed.12078.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/23506473" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1111%2Ftbed.12078" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR8">8. <span class="element-citation">Capriotti E, Fariselli P, Rossi I, Casadio R. A three-state prediction of single pointmutations on protein stability changes. <span><span class="ref-journal">BMC Bioinformatics. </span>2008;<span class="ref-vol">9</span>(2):S6. doi: 10.1186/1471-2105-9-S2-S6.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2323669/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/18387208" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1186%2F1471-2105-9-S2-S6" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR9">9. <span class="element-citation">Cilia E, Teso S, Ammendola S, Lenaerts T, Passerini A. Predicting virus mutations through statistical relational learning. <span><span class="ref-journal">BMC Bioinformatics. </span>2014;<span class="ref-vol">15</span>:309. doi: 10.1186/1471-2105-15-309.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4261881/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/25238967" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1186%2F1471-2105-15-309" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR10">10. <span class="element-citation">Lotfi M. Zare-Mirakabad F, Montaseri S, RNA secondary structure prediction based on SHAPE data in helix regions. <span><span class="ref-journal">J. Theor. Biol. </span>2015;<span class="ref-vol">380</span>:178–182. doi: 10.1016/j.jtbi.2015.05.026.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/26037307" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1016%2Fj.jtbi.2015.05.026" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR11">11. <span class="element-citation">Chang
 TH, Wu LC, Chen YT, Huang HD, Liu BJ, Cheng KF, Horng JT. 
Characterization and prediction of mRNA polyadenylation sites in human 
genes. <span><span class="ref-journal">Med. Biol. Eng. Comput. </span>2011;<span class="ref-vol">49</span>(4):463–72. doi: 10.1007/s11517-011-0732-4.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/21286831" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1007%2Fs11517-011-0732-4" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR12">12. <span class="element-citation">Kusy
 M, Obrzut B, Kluska J. Application of gene expression programming and 
neural networks to predict adverse events of radical hysterectomy in 
cervical cancer patients. <span><span class="ref-journal">Med. Biol. Eng. Comput. </span>2013;<span class="ref-vol">51</span>(12):1357–1365. doi: 10.1007/s11517-013-1108-8.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3825140/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/24136688" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1007%2Fs11517-013-1108-8" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR13">13. <span class="element-citation">Hobolth
 A. A Markov chain Monte Carlo expectation maximization algorithm for 
statistical analysis of DNA sequence evolution with neighbor-dependent 
substitution rates. <span><span class="ref-journal">J. Comput. Graph. Stat. </span>2008;<span class="ref-vol">17</span>(1):138–162. doi: 10.1198/106186008X289010.</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1198%2F106186008X289010" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR14">14. <span class="element-citation">Arndt PF, Hwa T. Identification and measurement of neighbor-dependent nucleotide substitution processes. <span><span class="ref-journal">Binformatics. </span>2005;<span class="ref-vol">21</span>(10):2322–2328. doi: 10.1093/bioinformatics/bti376.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/15769841" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1093%2Fbioinformatics%2Fbti376" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR15">15. <span class="element-citation">Ferguson NM, Anderson RM. Predicting evolutionary change in the influenza A virus. <span><span class="ref-journal">Nat. Med. </span>2002;<span class="ref-vol">8</span>:562–563. doi: 10.1038/nm0602-562.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/12042802" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1038%2Fnm0602-562" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR16">16. <span class="element-citation">Smith
 DJ, Lapedes AS, de Jong JC, Bestebroer TM, Rimmelzwaan GF. Mapping the 
antigenic and genetic evolution of influenza virus. <span><span class="ref-journal">Science. </span>2004;<span class="ref-vol">305</span>(5682):371–376. doi: 10.1126/science.1097211.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/15218094" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1126%2Fscience.1097211" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR17">17. <span class="element-citation">Choi
 K-S, Lee E-K, Jeon W-J, Kwon J-H, Lee J-H, Sung H-W. Molecular 
epidemiologic investigation of lentogenic Newcastle disease virus from 
domestic birds at live bird markets in Korea. <span><span class="ref-journal">Avian Dis. </span>2012;<span class="ref-vol">56</span>(1):218–223. doi: 10.1637/9699-030311-ResNote.1.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/22545549" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1637%2F9699-030311-ResNote.1" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR18">18. <span class="element-citation">Qin
 Z-M, Tan L-T, Xu H-Y, Ma B-C, Wang Y-L, Yuan X-Y, Liu W-J. Pathotypical
 characterization and molecular epidemiology of Newcastle disease virus 
isolates from different hosts in China from 1996 to 2005. <span><span class="ref-journal">J. Clin. Microbiol. </span>2008;<span class="ref-vol">46</span>(4):601–611. doi: 10.1128/JCM.01356-07.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2238121/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/18077643" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1128%2FJCM.01356-07" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR19">19. <span class="mixed-citation">Y
 Choi, GE Sims, S Murphy, JR Miller, AP Chan, Predicting the functional 
effect of amino acid substitutions and indels. PLoS One. <strong>7</strong>(10), e46688 (2012). doi:<a href="http://dx.doi.org/10.1371/journal.pone.0046688" ref="reftype=extlink&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=External%7CLink%7CURI&amp;rendering-type=normal" target="pmc_ext">http://dx.doi.org/10.1371/journal.pone.0046688</a>. <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3466303/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/23056405" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR20">20. <span class="element-citation">Wang D, Larder B. Enhanced prediction of lopinavir resistance from genotype by use of artificial neural networks. <span><span class="ref-journal">J. Infect. Dis. </span>2003;<span class="ref-vol">188</span>(11):653–660. doi: 10.1086/377453.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/12934180" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1086%2F377453" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR21">21. <span class="element-citation">Cao ZW, Han LY, Zheng CJ, Ji ZL, Chen X, Lin HH, Chen YZ. Computer prediction of drug resistance mutations in proteins. <span><span class="ref-journal">Drug Discov. Today. </span>2005;<span class="ref-vol">10</span>(7):521–529. doi: 10.1016/S1359-6446(05)03377-5.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/15809198" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1016%2FS1359-6446%2805%2903377-5" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR22">22. <span class="element-citation">Beerenwinkel
 N, Schmidt B, Walter H, Kaiser R, Lengauer T, Hoffmann D, Korn K, 
Selbig J. Diversity and complexity of HIV-1 drug resistance: a 
bioinformatics approach to predicting phenotype from genotype. <span><span class="ref-journal">Proc. Natl. Acad. Sci. USA. </span>1999;<span class="ref-vol">99</span>(12):8271–8276. doi: 10.1073/pnas.112177799.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC123057/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/12060770" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1073%2Fpnas.112177799" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR23">23. <span class="element-citation">Ding
 J, Bashashati A, Roth A, Oloumi A, Tse K, Zeng T, Haffari G, Hirst M, 
Marra MA, Condon A, Aparicio S, Shah SP. Feature-based classifiers for 
somatic mutation detection in tumour-normal paired sequencing data. <span><span class="ref-journal">Bioinformatics. </span>2012;<span class="ref-vol">28</span>(2):167–75. doi: 10.1093/bioinformatics/btr629.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3259434/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/22084253" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1093%2Fbioinformatics%2Fbtr629" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR24">24. <span class="element-citation">Lai D, Proctor JR, Meyer IM. On the importance of cotranscriptional RNA structure formation. <span><span class="ref-journal">RNA. </span>2013;<span class="ref-vol">19</span>(11):1461–1473. doi: 10.1261/rna.037390.112.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3851714/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/24131802" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1261%2Frna.037390.112" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR25">25. <span class="element-citation">Mathews DH, Moss WN, Turner DH. Folding and finding RNA secondary structure. <span><span class="ref-journal">Cold Spring Harb Perspect Biol. </span>2010;<span class="ref-vol">2</span>(12):a003665. doi: 10.1101/cshperspect.a003665.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2982177/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/20685845" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1101%2Fcshperspect.a003665" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR26">26. <span class="element-citation">Hofacker IL, Fekete M, Stadler PF. Secondary structure prediction for aligned RNA sequences. <span><span class="ref-journal">J. Mol. Biol. </span>2002;<span class="ref-vol">319</span>(5):1059–1066. doi: 10.1016/S0022-2836(02)00308-X.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/12079347" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1016%2FS0022-2836%2802%2900308-X" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR27">27. <span class="element-citation">Barash D, Churkin A. Mutational analysis in RNAs: comparing programs for RNA deleterious mutation prediction. <span><span class="ref-journal">Brief Bioinform. </span>2011;<span class="ref-vol">12</span>(2):104–114. doi: 10.1093/bib/bbq059.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/21422070" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1093%2Fbib%2Fbbq059" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR28">28. <span class="element-citation">Morin R, et al.  Profiling the HeLa S3 transcriptome using randomly primed cDNA and massively parallel short-read sequencing. <span><span class="ref-journal">BioTechniques. </span>2008;<span class="ref-vol">45</span>(1):81–94. doi: 10.2144/000112900.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/18611170" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.2144%2F000112900" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR29">29. <span class="element-citation">Goya
 R, Sun MG, Morin RD, Leung G, Ha G, Wiegand KC, Senz J, Crisan A, Marra
 MA, Hirst M, Huntsman D, Murphy KP, Aparicio S, Shah SP. SNVMix: 
predicting single nucleotide variants from next-generation sequencing of
 tumors. <span><span class="ref-journal">Bioinformatics. </span>2010;<span class="ref-vol">26</span>(6):730–736. doi: 10.1093/bioinformatics/btq040.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2832826/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/20130035" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1093%2Fbioinformatics%2Fbtq040" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR30">30. <span class="element-citation">Li H, Ruan J, Durbin R. Mapping short DNA sequencing reads and calling variants using mapping quality scores. <span><span class="ref-journal">Genome Res. </span>2008;<span class="ref-vol">18</span>:1851–1858. doi: 10.1101/gr.078212.108.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2577856/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/18714091" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1101%2Fgr.078212.108" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR31">31. <span class="element-citation">Berard J, Guéguen L. Accurate estimation of substitution rates with neighbor-dependent models in a phylogenetic context. <span><span class="ref-journal">J. Systmatic Biol. </span>2012;<span class="ref-vol">61</span>(3):510–21. doi: 10.1093/sysbio/sys024.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/22331438" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1093%2Fsysbio%2Fsys024" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR32">32. <span class="element-citation">Ke
 GM, Chuang KP, Chang CD, Lin MY, Liu HJ. Analysis of sequence and 
haemagglutinin activity of the HN glycoprotein of New-castle disease 
virus. <span><span class="ref-journal">Avian Pathol. </span>2010;<span class="ref-vol">39</span>(3):235–244. doi: 10.1080/03079451003789331.</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/20544431" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1080%2F03079451003789331" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR33">33. <span class="element-citation">Bal M. Rough sets theory as symbolic data mining method: an application on complete decision table. <span><span class="ref-journal">Inform. Sci. Lett. </span>2013;<span class="ref-vol">2</span>(1):35–47. doi: 10.12785/isl/020105.</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.12785%2Fisl%2F020105" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span></div><div class="ref-cit-blk half_rhythm" id="CR34">34. <span class="element-citation">Wang J-Y, Liu W-H, Ren J-J, Tang P, Wu N, Liu H-J. Complete genome sequence of a newly emerging Newcastle disease virus. <span><span class="ref-journal">Genome Announc. </span>2013;<span class="ref-vol">1</span>(3):196–13.</span> <span class="nowrap ref pmc">[<a class="int-reflink" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3650438/">PMC free article</a>]</span>  <span class="nowrap ref pubmed">[<a href="https://www.ncbi.nlm.nih.gov/pubmed/23661479" target="pmc_ext" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Entrez%7CPubMed%7CRecord&amp;rendering-type=normal">PubMed</a>]</span></span></div></div></div><div style="display: none; width: 200px; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true" class="ui-helper-reset ui-ncbipopper-wrapper ui-ncbilinksmenu"><ul id="ui-ncbiinpagenav-2"><li><a href="#Abs1title">Abstract</a></li><li><a href="#Sec1title">Introduction</a></li><li><a href="#Sec2title">Methods</a></li><li><a href="#Sec7title">Results</a></li><li><a href="#Sec10title">Discussion</a></li><li><a href="#Sec12title">Conclusions</a></li><li><a href="#__ackidm140672725254304title">Acknowledgements</a></li><li><a href="#__fn-groupidm140672725071728title">Footnotes</a></li><li><a href="#Bib1title">References</a></li></ul></div></div><!--post-content--><hr class="whole_rhythm no_bottom_margin"><div class="courtesy-note no_margin small">Articles from <span class="acknowledgment-journal-title">EURASIP Journal on Bioinformatics and Systems Biology</span> are provided here courtesy of <strong>Springer</strong></div></div>
            
            
        
            
        </div>
        <!-- Book content -->
    </div>
    
    <div id="rightcolumn" class="four_col col last">
        <!-- Custom content above discovery portlets -->
        <div class="col6">
            
        </div>
        
        <div xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"><div class="format-menu"><h2>Formats:</h2><ul><li class="selected">Article</li> | <li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/?report=reader">PubReader</a></li> | <li class="epub-link"><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/epub/">ePub (beta)</a></li> | <li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4867776/pdf/13637_2016_Article_42.pdf">PDF (2.5M)</a></li> | <li><a href="#" data-citationid="PMC4867776" class="citationexporter ctxp" role="button" aria-expanded="false" aria-haspopup="true">Citation</a></li></ul></div></div><div xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="share-buttons"><h2>Share</h2><ul><li class="facebook"><a href="https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC4867776%2F" role="button" aria-expanded="false" aria-haspopup="true" target="_blank"><img src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/4047626.png" alt="Share on Facebook">
                             Facebook
                        </a></li><li class="twitter"><a href="https://twitter.com/intent/tweet?url=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC4867776%2F&amp;text=The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques" role="button" aria-expanded="false" aria-haspopup="true" target="_blank"><img src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/4047627.png" alt="Share on Twitter">
                             Twitter
                        </a></li><li class="gplus"><a href="https://plus.google.com/share?url=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC4867776%2F" role="button" aria-expanded="false" aria-haspopup="true" target="_blank"><img src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/4047628.png" alt="Share on Google Plus">
                             Google+
                        </a></li></ul></div>
        
        <div id="ajax-portlets" data-pmid="27257410" data-aiid="5270457" data-aid="4867776" data-iid="285916" data-domainid="495" data-domain="ejbiosysbio" data-accid="PMC4867776" data-md5="bff3285f4f5ae60bcc341abf6d33a65e">
<div class="portlet pubmed_favoritesad">
<div class="portlet_head">
<div class="portlet_title"><h3><span>Save items</span></h3></div>
<a name="Shutter" sid="1" href="#" class="portlet_shutter ui-ncbitoggler-open" title="Show/hide content" remembercollapsed="true" pgsec_name="pmfavad" toggles="ui-portlet_content-4" role="button" aria-expanded="true" style="position: absolute; padding: 0px;"></a>
</div>
<div class="ui-helper-reset" aria-live="assertive"><div class="portlet_content ui-ncbitoggler ui-ncbitoggler-slave-open" id="ui-portlet_content-4" aria-hidden="false"><div xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
<input name="absid" id="absid" value="5270457" type="hidden"><div id="pubmed_favoritesad" class="empty" data-db="pmc">
<span role="menubar"><div class="ui-ncbisetswitch ltd-hover" role="menuitem" aria-expanded="false" aria-haspopup="true"><a href="https://www.ncbi.nlm.nih.gov/myncbi/collections/" class="jig-ncbisetswitch blind collink ltd-hover ui-ncbisetswitch-hasStar" id="favList" data-jigconfig="destSelector:'#favUL',starExists:true,createCollectionUrl:'#',manageCollectionsUrl:'/myncbi/collections/'"><span class="star"></span>Add to Favorites</a><a class="ui-ncbisetswitch-button ltd-hover" title="View more options" href="#"><span class="ui-icon ui-icon-triangle-1-s ui-ncbisetswitch-down">View more options</span></a></div></span><div style="display: none; width: 200px; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true" class="ui-helper-reset ui-ncbipopper-wrapper ui-ncbilinksmenu ui-ncbisetswitch-popper"><ul id="favUL" class="ui-ncbisetswitch-options"></ul><ul class="ui-ncbisetswitch-actions"><li class="ui-ncbisetswitch-actions-create" style="display: block;"><a class="ui-ncbisetswitch-create-collection" href="#">Create collection...</a></li><li class="ui-ncbisetswitch-actions-manage" style="display: block;"><a class="ui-ncbisetswitch-manage-collections" href="https://www.ncbi.nlm.nih.gov/myncbi/collections/">Manage collections...</a></li></ul></div>
</div>
<div class="colloading">loading</div>
<span id="submenu_AddToCollections"><form method="post"><input class="button_apply" name="addtocollection" value="1" type="submit"></form></span>
</div></div></div>
</div><div class="portlet">
<div class="portlet_head">
<div class="portlet_title"><h3><span>Similar articles in PubMed</span></h3></div>
<a name="Shutter" sid="1" href="#" class="portlet_shutter ui-ncbitoggler-open" title="Show/hide content" remembercollapsed="true" pgsec_name="PBooksDiscovery_RA" toggles="ui-portlet_content-5" role="button" aria-expanded="true" style="position: absolute; padding: 0px;"></a>
</div>
<div class="ui-helper-reset" aria-live="assertive"><div class="portlet_content ui-ncbitoggler ui-ncbitoggler-slave-open" id="ui-portlet_content-5" aria-hidden="false">
<ul>
<li class="brieflinkpopper two_line">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pubmed/24246731" ref="reftype=relart&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|RelatedArticles&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal" role="button" aria-expanded="false" aria-haspopup="true">Comparison of Multiple Linear Regressions and Neural Networks based QSAR models for the design of new antitubercular compounds.</a><span class="source">[Eur J Med Chem. 2013]</span><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-6" aria-live="assertive" aria-hidden="true" style="top: -100px; left: -100px; display: none;">
<em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="author">Ventura C, Latino DA, Martins F. </em><em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="cit">Eur J Med Chem. 2013; 70:831-45. Epub 2013 Oct 23.</em>
</div>
</li>
<li class="brieflinkpopper two_line">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pubmed/27965630" ref="reftype=relart&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|RelatedArticles&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal" role="button" aria-expanded="false" aria-haspopup="true">Evaluation
 of Machine Learning and Rules-Based Approaches for Predicting 
Antimicrobial Resistance Profiles in Gram-negative Bacilli from Whole 
Genome Sequence Data.</a><span class="source">[Front Microbiol. 2016]</span><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-7" aria-live="assertive" aria-hidden="true" style="top: -100px; left: -100px; display: none;">
<em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="author">Pesesky MW, Hussain T, Wallace M, Patel S, Andleeb S, Burnham CD, Dantas G. </em><em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="cit">Front Microbiol. 2016; 7:1887. Epub 2016 Nov 28.</em>
</div>
</li>
<li class="brieflinkpopper two_line">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pubmed/25150834" ref="reftype=relart&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|RelatedArticles&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal" role="button" aria-expanded="false" aria-haspopup="true">Prediction
 of hepatitis C virus interferon/ribavirin therapy outcome based on 
viral nucleotide attributes using machine learning algorithms.</a><span class="source">[BMC Res Notes. 2014]</span><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-8" aria-live="assertive" aria-hidden="true" style="top: -100px; left: -100px; display: none;">
<em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="author">KayvanJoo AH, Ebrahimi M, Haqshenas G. </em><em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="cit">BMC Res Notes. 2014 Aug 23; 7:565. Epub 2014 Aug 23.</em>
</div>
</li>
<li class="brieflinkpopper two_line">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pubmed/14626050" ref="reftype=relart&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|RelatedArticles&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal" role="button" aria-expanded="false" aria-haspopup="true">Molecular biological assessment methods and understanding the course of the HIV infection.</a><span class="source">[APMIS Suppl. 2003]</span><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-9" aria-live="assertive" aria-hidden="true" style="top: -100px; left: -100px; display: none;">
<em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="author">Katzenstein TL. </em><em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="cit">APMIS Suppl. 2003; (114):1-37. </em>
</div>
</li>
<li class="brieflinkpopper two_line">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pubmed/26042329" ref="reftype=relart&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|RelatedArticles&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal" role="button" aria-expanded="false" aria-haspopup="true">Engineering Aspects of Olfaction</a><span class="source">[Neuromorphic Olfaction. 2013]</span><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-10" aria-live="assertive" aria-hidden="true" style="top: -100px; left: -100px; display: none;">
<em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="author">Persaud KC. </em><em xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" class="cit">Neuromorphic Olfaction. 2013</em>
</div>
</li>
</ul>
<a class="seemore" href="https://www.ncbi.nlm.nih.gov/sites/entrez?db=pubmed&amp;cmd=link&amp;linkname=pubmed_pubmed_reviews&amp;uid=27257410&amp;log%24=relatedreviews&amp;logdbfrom=pmc" ref="reftype=relart&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|RelatedArticles&amp;TO=Entrez|Pubmed|Reviews&amp;rendering-type=normal">See reviews...</a><a class="seemore" href="https://www.ncbi.nlm.nih.gov/sites/entrez?db=pubmed&amp;cmd=link&amp;linkname=pubmed_pubmed&amp;uid=27257410&amp;log%24=relatedarticles&amp;logdbfrom=pmc" ref="reftype=relart&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|RelatedArticles&amp;TO=Entrez|Pubmed|Related%20Records&amp;rendering-type=normal">See all...</a>
</div></div>
</div><div class="portlet">
<div class="portlet_head">
<div class="portlet_title"><h3><span>Links</span></h3></div>
<a name="Shutter" sid="1" href="#" class="portlet_shutter ui-ncbitoggler-open" title="Show/hide content" remembercollapsed="true" pgsec_name="PMCVCitedByPmcArticlesP" toggles="ui-portlet_content-11" role="button" aria-expanded="true" style="position: absolute; padding: 0px;"></a>
</div>
<div class="ui-helper-reset" aria-live="assertive"><div class="portlet_content ui-ncbitoggler ui-ncbitoggler-slave-open" id="ui-portlet_content-11" aria-hidden="false"><ul>
<li class="brieflinkpopper">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pmc/?Db=nuccore&amp;DbFrom=pmc&amp;Cmd=Link&amp;LinkName=pmc_nuccore&amp;IdsFromResult=4867776" ref="reftype=Nucleotide&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|DiscoveryLinks&amp;TO=Entrez|Crosslink|Nucleotide&amp;rendering-type=normal" role="button" aria-expanded="false" aria-haspopup="true">Nucleotide</a><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-12" aria-live="assertive" aria-hidden="true" style="top: -100px; left: -100px; display: none;">Nucleotide<div class="brieflinkpopdesc">Published Nucleotide sequences</div>
</div>
</li>
<li class="brieflinkpopper">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pubmed/27257410/" ref="reftype=PubMed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|DiscoveryLinks&amp;TO=Entrez|PubMed|Record&amp;rendering-type=normal" role="button" aria-expanded="false" aria-haspopup="true">PubMed</a><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-13" aria-live="assertive" aria-hidden="true" style="top: -100px; left: -100px; display: none;">PubMed<div class="brieflinkpopdesc">PubMed citations for these articles</div>
</div>
</li>
<li class="brieflinkpopper">
<a class="brieflinkpopperctrl" href="https://www.ncbi.nlm.nih.gov/pmc/?Db=taxonomy&amp;DbFrom=pmc&amp;Cmd=Link&amp;LinkName=pmc_taxonomy&amp;IdsFromResult=4867776" ref="reftype=Taxonomy&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|DiscoveryLinks&amp;TO=Entrez|Crosslink|Taxonomy&amp;rendering-type=normal" role="button" aria-expanded="false" aria-haspopup="true">Taxonomy</a><div class="brieflinkpop ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" id="ui-brieflinkpop-14" aria-live="assertive" aria-hidden="true" style="top: -100px; left: -100px; display: none;">Taxonomy<div class="brieflinkpopdesc">Related taxonomy entry</div>
</div>
</li>
</ul></div></div>
</div><div class="portlet">
<div class="portlet_head">
<div class="portlet_title"><h3><span>Recent Activity</span></h3></div>
<a name="Shutter" sid="1" href="#" class="portlet_shutter ui-ncbitoggler-open" title="Show/hide content" remembercollapsed="true" pgsec_name="recent_activity" toggles="ui-portlet_content-15" role="button" aria-expanded="true" style="position: absolute; padding: 0px;"></a>
</div>
<div class="ui-helper-reset" aria-live="assertive"><div class="portlet_content ui-ncbitoggler ui-ncbitoggler-slave-open" id="ui-portlet_content-15" aria-hidden="false"><div xmlns:np="http://ncbi.gov/portal/XSLT/namespace" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="HTDisplay" class="">
<div class="action">
<a href="javascript:historyDisplayState('ClearHT')">Clear</a><a href="javascript:historyDisplayState('HTOff')" class="HTOn">Turn Off</a><a href="javascript:historyDisplayState('HTOn')" class="HTOff">Turn On</a>
</div>
<ul id="activity">
<li class="ra_rcd ralinkpopper two_line">
<a class="htb ralinkpopperctrl" ref="log$=activity&amp;linkpos=1" href="https://www.ncbi.nlm.nih.gov/portal/utils/pageresolver.fcgi?recordid=594c8bc906203355f58e6cff">The prediction of virus mutation using neural networks and rough set techniques</a><div class="ralinkpop offscreen_noflow">The prediction of virus mutation using neural networks and rough set techniques<div class="brieflinkpopdesc">EURASIP Journal on Bioinformatics and Systems Biology. 2016 Dec; 2016()</div>
</div>
<div class="tertiary"></div>
</li>
<li class="ra_rcd ralinkpopper two_line">
<a class="htb ralinkpopperctrl" ref="log$=activity&amp;linkpos=2" href="https://www.ncbi.nlm.nih.gov/portal/utils/pageresolver.fcgi?recordid=594c7c1a06203355f58c7760">Influenza A virus (A/Aragon/RR3207/2008(H1N1)) segment 4 hemagglutinin (HA) gene...</a><div class="ralinkpop offscreen_noflow">Influenza A virus (A/Aragon/RR3207/2008(H1N1)) segment 4 hemagglutinin (HA) gene, partial cds<div class="brieflinkpopdesc">gi|254597362|gb|GQ397633.1|</div>
</div>
<div class="tertiary">Nucleotide</div>
</li>
<li class="ra_rcd ralinkpopper two_line">
<a class="htb ralinkpopperctrl" ref="log$=activity&amp;linkpos=3" href="https://www.ncbi.nlm.nih.gov/portal/utils/pageresolver.fcgi?recordid=594c7bb606203355f58c69e0">Influenza A virus (A/Ankara-TR/170/2008(H1N1)) neuraminidase (NA) gene, partial ...</a><div class="ralinkpop offscreen_noflow">Influenza A virus (A/Ankara-TR/170/2008(H1N1)) neuraminidase (NA) gene, partial cds<div class="brieflinkpopdesc">gi|188526740|gb|EU681263.1|</div>
</div>
<div class="tertiary">Nucleotide</div>
</li>
<li class="ra_rcd ralinkpopper two_line">
<a class="htb ralinkpopperctrl" ref="log$=activity&amp;linkpos=4" href="https://www.ncbi.nlm.nih.gov/portal/utils/pageresolver.fcgi?recordid=594c7b57758fd9a6fb4d552f">Influenza A virus (A/Anhui-Baohe/1514/2008(H1N1)) segment 7 sequence</a><div class="ralinkpop offscreen_noflow">Influenza A virus (A/Anhui-Baohe/1514/2008(H1N1)) segment 7 sequence<div class="brieflinkpopdesc">gi|293598898|gnl|CNIC|CNIC-H1N1-477 |CY059257.1|</div>
</div>
<div class="tertiary">Nucleotide</div>
</li>
<li class="ra_rcd ralinkpopper two_line">
<a class="htb ralinkpopperctrl" ref="log$=activity&amp;linkpos=5" href="https://www.ncbi.nlm.nih.gov/portal/utils/pageresolver.fcgi?recordid=594c7afa06203355f58c515f">Influenza A virus (A/Alaska/05/2008(H1N1)) segment 7 matrix protein 2 (M2) and m...</a><div class="ralinkpop offscreen_noflow">Influenza A virus (A/Alaska/05/2008(H1N1)) segment 7 matrix protein 2 (M2) and matrix protein 1 (M1) genes, complete cds<div class="brieflinkpopdesc">gi|255960228|gb|GQ475819.1|</div>
</div>
<div class="tertiary">Nucleotide</div>
</li>
</ul>
<p class="HTOn">Your browsing activity is empty.</p>
<p class="HTOff">Activity recording is turned off.</p>
<p id="turnOn" class="HTOff"><a href="javascript:historyDisplayState('HTOn')">Turn recording back on</a></p>
<a class="seemore" href="https://www.ncbi.nlm.nih.gov/sites/myncbi/recentactivity">See more...</a>
</div></div></div>
</div>
<div>
<div class="portlet brieflink pmc_para_cit" id="crb--__p2" name="crb--__p2" rid="__p2" style="position: absolute; visibility: visible; width: 370px; top: 845px; display: none;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-16"><ul>
<li class="two_line" reference_id="CR1">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/16140732/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal"><span class="invert">Review</span> Adaptive value of high mutation rates of RNA viruses: separating causes from consequences.</a><span class="one_line_source">[J Virol.  2005]</span><div class="alt-note">
<div class="authors">Elena SF, Sanjuán R</div>
<div class="citation">J Virol. 2005 Sep; 79(18):11555-8.</div>
</div>
</li>
<li class="two_line" reference_id="CR2">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/26578204/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal"><span class="invert">Review</span> The population genetics of drug resistance evolution in natural populations of viral, bacterial and eukaryotic pathogens.</a><span class="one_line_source">[Mol Ecol.  2016]</span><div class="alt-note">
<div class="authors">Wilson BA, Garud NR, Feder AF, Assaf ZJ, Pennings PS</div>
<div class="citation">Mol Ecol. 2016 Jan; 25(1):42-66.</div>
</div>
</li>
<li class="two_line" reference_id="CR3">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/23325689/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">T-705 (favipiravir) induces lethal mutagenesis in influenza A H1N1 viruses in vitro.</a><span class="one_line_source">[J Virol.  2013]</span><div class="alt-note">
<div class="authors">Baranovich T, Wong SS, Armstrong J, Marjuki H, Webby RJ, Webster RG, Govorkova EA</div>
<div class="citation">J Virol. 2013 Apr; 87(7):3741-51.</div>
</div>
</li>
</ul></div>
</div>
<div class="portlet brieflink pmc_para_cit" id="crb--__p3" name="crb--__p3" rid="__p3" style="position: absolute; visibility: visible; width: 370px; top: 1253px;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-17"><ul>
<li class="two_line" reference_id="CR5">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/16848976/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal"><span class="invert">Review</span> From DNA to RNA to disease and back: the 'central dogma' of regulatory disease variation.</a><span class="one_line_source">[Hum Genomics.  2006]</span><div class="alt-note">
<div class="authors">Stranger BE, Dermitzakis ET</div>
<div class="citation">Hum Genomics. 2006 Jun; 2(6):383-90.</div>
</div>
</li>
<li class="two_line" reference_id="CR6">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/18846087/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Next-generation DNA sequencing.</a><span class="one_line_source">[Nat Biotechnol.  2008]</span><div class="alt-note">
<div class="authors">Shendure J, Ji H</div>
<div class="citation">Nat Biotechnol. 2008 Oct; 26(10):1135-45.</div>
</div>
</li>
<li class="two_line" reference_id="CR7">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/23506473/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Phylogenetic analysis of canine parvovirus isolates from Sichuan and Gansu provinces of China in 2011.</a><span class="one_line_source">[Transbound Emerg Dis.  2015]</span><div class="alt-note">
<div class="authors">Xu J, Guo HC, Wei YQ, Shu L, Wang J, Li JS, Cao SZ, Sun SQ</div>
<div class="citation">Transbound Emerg Dis. 2015 Feb; 62(1):91-5.</div>
</div>
</li>
<li class="two_line" reference_id="CR8">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/18387208/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">A three-state prediction of single point mutations on protein stability changes.</a><span class="one_line_source">[BMC Bioinformatics.  2008]</span><div class="alt-note">
<div class="authors">Capriotti E, Fariselli P, Rossi I, Casadio R</div>
<div class="citation">BMC Bioinformatics. 2008 Mar 26; 9 Suppl 2():S6.</div>
</div>
</li>
<li class="two_line" reference_id="CR9">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/25238967/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Predicting virus mutations through statistical relational learning.</a><span class="one_line_source">[BMC Bioinformatics.  2014]</span><div class="alt-note">
<div class="authors">Cilia E, Teso S, Ammendola S, Lenaerts T, Passerini A</div>
<div class="citation">BMC Bioinformatics. 2014 Sep 19; 15():309.</div>
</div>
</li>
<li class="two_line" reference_id="CR10">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/26037307/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">RNA secondary structure prediction based on SHAPE data in helix regions.</a><span class="one_line_source">[J Theor Biol.  2015]</span><div class="alt-note">
<div class="authors">Lotfi M, Zare-Mirakabad F, Montaseri S</div>
<div class="citation">J Theor Biol. 2015 Sep 7; 380():178-82.</div>
</div>
</li>
<li class="two_line" reference_id="CR12">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/24136688/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Application
 of gene expression programming and neural networks to predict adverse 
events of radical hysterectomy in cervical cancer patients.</a><span class="one_line_source">[Med Biol Eng Comput.  2013]</span><div class="alt-note">
<div class="authors">Kusy M, Obrzut B, Kluska J</div>
<div class="citation">Med Biol Eng Comput. 2013 Dec; 51(12):1357-65.</div>
</div>
</li>
<li class="two_line" reference_id="CR14">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/15769841/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Identification and measurement of neighbor-dependent nucleotide substitution processes.</a><span class="one_line_source">[Bioinformatics.  2005]</span><div class="alt-note">
<div class="authors">Arndt PF, Hwa T</div>
<div class="citation">Bioinformatics. 2005 May 15; 21(10):2322-8.</div>
</div>
</li>
</ul></div>
</div>
<div class="portlet brieflink pmc_para_cit" id="crb--__p4" name="crb--__p4" rid="__p4" style="position: absolute; visibility: visible; width: 370px; top: 1704px;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-18"><ul>
<li class="two_line" reference_id="CR15">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/12042802/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Predicting evolutionary change in the influenza A virus.</a><span class="one_line_source">[Nat Med.  2002]</span><div class="alt-note">
<div class="authors">Ferguson NM, Anderson RM</div>
<div class="citation">Nat Med. 2002 Jun; 8(6):562-3.</div>
</div>
</li>
<li class="two_line" reference_id="CR16">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/15218094/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Mapping the antigenic and genetic evolution of influenza virus.</a><span class="one_line_source">[Science.  2004]</span><div class="alt-note">
<div class="authors">Smith DJ, Lapedes AS, de Jong JC, Bestebroer TM, Rimmelzwaan GF, Osterhaus AD, Fouchier RA</div>
<div class="citation">Science. 2004 Jul 16; 305(5682):371-6.</div>
</div>
</li>
</ul></div>
</div>
<div class="portlet brieflink pmc_para_cit" id="crb--__p5" name="crb--__p5" rid="__p5" style="position: absolute; visibility: visible; width: 370px; top: 2199px;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-19"><ul>
<li class="two_line" reference_id="CR17">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/22545549/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Molecular epidemiologic investigation of lentogenic Newcastle disease virus from domestic birds at live bird markets in Korea.</a><span class="one_line_source">[Avian Dis.  2012]</span><div class="alt-note">
<div class="authors">Choi KS, Lee EK, Jeon WJ, Kwon JH, Lee JH, Sung HW</div>
<div class="citation">Avian Dis. 2012 Mar; 56(1):218-23.</div>
</div>
</li>
<li class="two_line" reference_id="CR18">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/18077643/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Pathotypical
 characterization and molecular epidemiology of Newcastle disease virus 
isolates from different hosts in China from 1996 to 2005.</a><span class="one_line_source">[J Clin Microbiol.  2008]</span><div class="alt-note">
<div class="authors">Qin ZM, Tan LT, Xu HY, Ma BC, Wang YL, Yuan XY, Liu WJ</div>
<div class="citation">J Clin Microbiol. 2008 Feb; 46(2):601-11.</div>
</div>
</li>
</ul></div>
</div>
<div class="portlet brieflink pmc_para_cit" id="crb--__p6" name="crb--__p6" rid="__p6" style="position: absolute; visibility: visible; width: 370px; top: 2534px;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-20"><ul>
<li class="two_line" reference_id="CR19">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/23056405/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Predicting the functional effect of amino acid substitutions and indels.</a><span class="one_line_source">[PLoS One.  2012]</span><div class="alt-note">
<div class="authors">Choi Y, Sims GE, Murphy S, Miller JR, Chan AP</div>
<div class="citation">PLoS One. 2012; 7(10):e46688.</div>
</div>
</li>
<li class="two_line" reference_id="CR20">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/12934180/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Enhanced prediction of lopinavir resistance from genotype by use of artificial neural networks.</a><span class="one_line_source">[J Infect Dis.  2003]</span><div class="alt-note">
<div class="authors">Wang D, Larder B</div>
<div class="citation">J Infect Dis. 2003 Sep 1; 188(5):653-60.</div>
</div>
</li>
<li class="two_line" reference_id="CR21">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/15809198/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal"><span class="invert">Review</span> Computer prediction of drug resistance mutations in proteins.</a><span class="one_line_source">[Drug Discov Today.  2005]</span><div class="alt-note">
<div class="authors">Cao ZW, Han LY, Zheng CJ, Ji ZL, Chen X, Lin HH, Chen YZ</div>
<div class="citation">Drug Discov Today. 2005 Apr 1; 10(7):521-9.</div>
</div>
</li>
<li class="two_line" reference_id="CR22">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/12060770/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Diversity and complexity of HIV-1 drug resistance: a bioinformatics approach to predicting phenotype from genotype.</a><span class="one_line_source">[Proc Natl Acad Sci U S A.  2002]</span><div class="alt-note">
<div class="authors">Beerenwinkel N, Schmidt B, Walter H, Kaiser R, Lengauer T, Hoffmann D, Korn K, Selbig J</div>
<div class="citation">Proc Natl Acad Sci U S A. 2002 Jun 11; 99(12):8271-6.</div>
</div>
</li>
<li class="two_line" reference_id="CR23">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/22084253/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Feature-based classifiers for somatic mutation detection in tumour-normal paired sequencing data.</a><span class="one_line_source">[Bioinformatics.  2012]</span><div class="alt-note">
<div class="authors">Ding J, Bashashati A, Roth A, Oloumi A, Tse K, Zeng T, Haffari G, Hirst M, Marra MA, Condon A, Aparicio S, Shah SP</div>
<div class="citation">Bioinformatics. 2012 Jan 15; 28(2):167-75.</div>
</div>
</li>
</ul></div>
</div>
<div class="portlet brieflink pmc_para_cit" id="crb--__p7" name="crb--__p7" rid="__p7" style="position: absolute; visibility: visible; width: 370px; top: 3051px;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-21"><ul>
<li class="two_line" reference_id="CR24">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/24131802/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal"><span class="invert">Review</span> On the importance of cotranscriptional RNA structure formation.</a><span class="one_line_source">[RNA.  2013]</span><div class="alt-note">
<div class="authors">Lai D, Proctor JR, Meyer IM</div>
<div class="citation">RNA. 2013 Nov; 19(11):1461-73.</div>
</div>
</li>
<li class="two_line" reference_id="CR25">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/20685845/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal"><span class="invert">Review</span> Folding and finding RNA secondary structure.</a><span class="one_line_source">[Cold Spring Harb Perspect Biol.  2010]</span><div class="alt-note">
<div class="authors">Mathews DH, Moss WN, Turner DH</div>
<div class="citation">Cold Spring Harb Perspect Biol. 2010 Dec; 2(12):a003665.</div>
</div>
</li>
<li class="two_line" reference_id="CR26">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/12079347/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Secondary structure prediction for aligned RNA sequences.</a><span class="one_line_source">[J Mol Biol.  2002]</span><div class="alt-note">
<div class="authors">Hofacker IL, Fekete M, Stadler PF</div>
<div class="citation">J Mol Biol. 2002 Jun 21; 319(5):1059-66.</div>
</div>
</li>
<li class="two_line" reference_id="CR27">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/21422070/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal"><span class="invert">Review</span> Mutational analysis in RNAs: comparing programs for RNA deleterious mutation prediction.</a><span class="one_line_source">[Brief Bioinform.  2011]</span><div class="alt-note">
<div class="authors">Barash D, Churkin A</div>
<div class="citation">Brief Bioinform. 2011 Mar; 12(2):104-14.</div>
</div>
</li>
</ul></div>
</div>
<div class="portlet brieflink pmc_para_cit" id="crb--__p8" name="crb--__p8" rid="__p8" style="position: absolute; visibility: visible; width: 370px; top: 3348px;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-22"><ul>
<li class="two_line" reference_id="CR6">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/18846087/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Next-generation DNA sequencing.</a><span class="one_line_source">[Nat Biotechnol.  2008]</span><div class="alt-note">
<div class="authors">Shendure J, Ji H</div>
<div class="citation">Nat Biotechnol. 2008 Oct; 26(10):1135-45.</div>
</div>
</li>
<li class="two_line" reference_id="CR28">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/18611170/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Profiling the HeLa S3 transcriptome using randomly primed cDNA and massively parallel short-read sequencing.</a><span class="one_line_source">[Biotechniques.  2008]</span><div class="alt-note">
<div class="authors">Morin R, Bainbridge M, Fejes A, Hirst M, Krzywinski M, Pugh T, McDonald H, Varhol R, Jones S, Marra M</div>
<div class="citation">Biotechniques. 2008 Jul; 45(1):81-94.</div>
</div>
</li>
</ul></div>
</div>
<div class="portlet brieflink pmc_para_cit" id="crb--__p9" name="crb--__p9" rid="__p9" style="position: absolute; visibility: visible; width: 370px; top: 3625px;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-23"><ul>
<li class="two_line" reference_id="CR29">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/20130035/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">SNVMix: predicting single nucleotide variants from next-generation sequencing of tumors.</a><span class="one_line_source">[Bioinformatics.  2010]</span><div class="alt-note">
<div class="authors">Goya R, Sun MG, Morin RD, Leung G, Ha G, Wiegand 
KC, Senz J, Crisan A, Marra MA, Hirst M, Huntsman D, Murphy KP, Aparicio
 S, Shah SP</div>
<div class="citation">Bioinformatics. 2010 Mar 15; 26(6):730-6.</div>
</div>
</li>
<li class="two_line" reference_id="CR30">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/18714091/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Mapping short DNA sequencing reads and calling variants using mapping quality scores.</a><span class="one_line_source">[Genome Res.  2008]</span><div class="alt-note">
<div class="authors">Li H, Ruan J, Durbin R</div>
<div class="citation">Genome Res. 2008 Nov; 18(11):1851-8.</div>
</div>
</li>
</ul></div>
</div>
<div class="portlet brieflink pmc_para_cit" id="crb--__p10" name="crb--__p10" rid="__p10" style="position: absolute; visibility: visible; width: 370px; top: 3935px;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-24"><ul><li class="two_line" reference_id="CR30">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/18714091/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Mapping short DNA sequencing reads and calling variants using mapping quality scores.</a><span class="one_line_source">[Genome Res.  2008]</span><div class="alt-note">
<div class="authors">Li H, Ruan J, Durbin R</div>
<div class="citation">Genome Res. 2008 Nov; 18(11):1851-8.</div>
</div>
</li></ul></div>
</div>
<div class="portlet brieflink pmc_para_cit" id="crb--__p13" name="crb--__p13" rid="__p13" style="position: absolute; visibility: visible; width: 370px; top: 4244px;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-25"><ul><li class="two_line" reference_id="CR31">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/22331438/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Accurate estimation of substitution rates with neighbor-dependent models in a phylogenetic context.</a><span class="one_line_source">[Syst Biol.  2012]</span><div class="alt-note">
<div class="authors">Bérard J, Guéguen L</div>
<div class="citation">Syst Biol. 2012 May; 61(3):510-21.</div>
</div>
</li></ul></div>
</div>
<div class="portlet brieflink pmc_para_cit" id="crb--__p14" name="crb--__p14" rid="__p14" style="position: absolute; visibility: visible; width: 370px; top: 4365px;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-26"><ul><li class="two_line" reference_id="CR32">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/20544431/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Analysis of sequence and haemagglutinin activity of the HN glycoprotein of Newcastle disease virus.</a><span class="one_line_source">[Avian Pathol.  2010]</span><div class="alt-note">
<div class="authors">Ke GM, Chuang KP, Chang CD, Lin MY, Liu HJ</div>
<div class="citation">Avian Pathol. 2010 Jun; 39(3):235-44.</div>
</div>
</li></ul></div>
</div>
<div class="portlet brieflink pmc_para_cit" id="crb--__p15" name="crb--__p15" rid="__p15" style="position: absolute; visibility: visible; width: 370px; top: 4504px;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-27"><ul>
<li class="two_line" reference_id="CR17">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/22545549/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Molecular epidemiologic investigation of lentogenic Newcastle disease virus from domestic birds at live bird markets in Korea.</a><span class="one_line_source">[Avian Dis.  2012]</span><div class="alt-note">
<div class="authors">Choi KS, Lee EK, Jeon WJ, Kwon JH, Lee JH, Sung HW</div>
<div class="citation">Avian Dis. 2012 Mar; 56(1):218-23.</div>
</div>
</li>
<li class="two_line" reference_id="CR18">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/18077643/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Pathotypical
 characterization and molecular epidemiology of Newcastle disease virus 
isolates from different hosts in China from 1996 to 2005.</a><span class="one_line_source">[J Clin Microbiol.  2008]</span><div class="alt-note">
<div class="authors">Qin ZM, Tan LT, Xu HY, Ma BC, Wang YL, Yuan XY, Liu WJ</div>
<div class="citation">J Clin Microbiol. 2008 Feb; 46(2):601-11.</div>
</div>
</li>
</ul></div>
</div>
<div class="portlet brieflink pmc_para_cit" id="crb--__p27" name="crb--__p27" rid="__p27" style="position: absolute; visibility: visible; width: 370px; top: 8015px;">
<div class="portlet_head"></div>
<div class="portlet_content" id="ui-portlet_content-28"><ul>
<li class="two_line" reference_id="CR17">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/22545549/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Molecular epidemiologic investigation of lentogenic Newcastle disease virus from domestic birds at live bird markets in Korea.</a><span class="one_line_source">[Avian Dis.  2012]</span><div class="alt-note">
<div class="authors">Choi KS, Lee EK, Jeon WJ, Kwon JH, Lee JH, Sung HW</div>
<div class="citation">Avian Dis. 2012 Mar; 56(1):218-23.</div>
</div>
</li>
<li class="two_line" reference_id="CR18">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/18077643/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Pathotypical
 characterization and molecular epidemiology of Newcastle disease virus 
isolates from different hosts in China from 1996 to 2005.</a><span class="one_line_source">[J Clin Microbiol.  2008]</span><div class="alt-note">
<div class="authors">Qin ZM, Tan LT, Xu HY, Ma BC, Wang YL, Yuan XY, Liu WJ</div>
<div class="citation">J Clin Microbiol. 2008 Feb; 46(2):601-11.</div>
</div>
</li>
<li class="two_line" reference_id="CR34">
<a href="https://www.ncbi.nlm.nih.gov/pubmed/23661479/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">Complete genome sequence of a newly emerging newcastle disease virus.</a><span class="one_line_source">[Genome Announc.  2013]</span><div class="alt-note">
<div class="authors">Wang JY, Liu WH, Ren JJ, Tang P, Wu N, Liu HJ</div>
<div class="citation">Genome Announc. 2013 May 9; 1(3):.</div>
</div>
</li>
</ul></div>
</div>
</div>   
    </div>
                
        <!-- Custom content below discovery portlets -->
        <div class="col7">
            
        </div>
    </div>
</div>

<!-- Custom content after all -->
<div class="col8">
    
</div>
<div class="col9">
    
</div>

<script src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/jquery_002.js"></script>
<script>
    (function($){
        $('.skiplink').each(function(i, item){
            var href = $($(item).attr('href'));
            href.attr('tabindex', '-1').addClass('skiptarget'); // ensure the target can receive focus
            $(item).on('click', function(event){
                event.preventDefault();
                $.scrollTo(href, 0, {
                    onAfter: function(){
                        href.focus();
                    }
                });
            });
        });
    })(jQuery);
</script>



<div id="body-link-poppers"><div id="body-link-popper-CR1" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true"><span class="invert">Review</span> Adaptive value of high mutation rates of RNA viruses: separating causes from consequences.<div class="authors">Elena SF, Sanjuán R</div><div class="citation">J Virol. 2005 Sep; 79(18):11555-8.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/16140732/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR1">Ref list</a>]</p></div><div id="body-link-popper-CR2" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true"><span class="invert">Review</span> The population genetics of drug resistance evolution in natural populations of viral, bacterial and eukaryotic pathogens.<div class="authors">Wilson BA, Garud NR, Feder AF, Assaf ZJ, Pennings PS</div><div class="citation">Mol Ecol. 2016 Jan; 25(1):42-66.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/26578204/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR2">Ref list</a>]</p></div><div id="body-link-popper-CR3" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">T-705 (favipiravir) induces lethal mutagenesis in influenza A H1N1 viruses in vitro.<div class="authors">Baranovich T, Wong SS, Armstrong J, Marjuki H, Webby RJ, Webster RG, Govorkova EA</div><div class="citation">J Virol. 2013 Apr; 87(7):3741-51.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/23325689/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR3">Ref list</a>]</p></div><div id="body-link-popper-CR4" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">4. <span class="element-citation">Loewe L. Genetic mutation. <span><span class="ref-journal">Nat. Educ. </span>2008;<span class="ref-vol">1</span>(1):113.</span></span> [<a href="#CR4">Ref list</a>]</div><div id="body-link-popper-CR5" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true"><span class="invert">Review</span> From DNA to RNA to disease and back: the 'central dogma' of regulatory disease variation.<div class="authors">Stranger BE, Dermitzakis ET</div><div class="citation">Hum Genomics. 2006 Jun; 2(6):383-90.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/16848976/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR5">Ref list</a>]</p></div><div id="body-link-popper-CR6" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Next-generation DNA sequencing.<div class="authors">Shendure J, Ji H</div><div class="citation">Nat Biotechnol. 2008 Oct; 26(10):1135-45.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/18846087/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR6">Ref list</a>]</p></div><div id="body-link-popper-CR7" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Phylogenetic analysis of canine parvovirus isolates from Sichuan and Gansu provinces of China in 2011.<div class="authors">Xu J, Guo HC, Wei YQ, Shu L, Wang J, Li JS, Cao SZ, Sun SQ</div><div class="citation">Transbound Emerg Dis. 2015 Feb; 62(1):91-5.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/23506473/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR7">Ref list</a>]</p></div><div id="body-link-popper-CR8" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">A three-state prediction of single point mutations on protein stability changes.<div class="authors">Capriotti E, Fariselli P, Rossi I, Casadio R</div><div class="citation">BMC Bioinformatics. 2008 Mar 26; 9 Suppl 2():S6.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/18387208/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR8">Ref list</a>]</p></div><div id="body-link-popper-CR9" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Predicting virus mutations through statistical relational learning.<div class="authors">Cilia E, Teso S, Ammendola S, Lenaerts T, Passerini A</div><div class="citation">BMC Bioinformatics. 2014 Sep 19; 15():309.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/25238967/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR9">Ref list</a>]</p></div><div id="body-link-popper-CR10" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">RNA secondary structure prediction based on SHAPE data in helix regions.<div class="authors">Lotfi M, Zare-Mirakabad F, Montaseri S</div><div class="citation">J Theor Biol. 2015 Sep 7; 380():178-82.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/26037307/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR10">Ref list</a>]</p></div><div id="body-link-popper-CR12" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Application
 of gene expression programming and neural networks to predict adverse 
events of radical hysterectomy in cervical cancer patients.<div class="authors">Kusy M, Obrzut B, Kluska J</div><div class="citation">Med Biol Eng Comput. 2013 Dec; 51(12):1357-65.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/24136688/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR12">Ref list</a>]</p></div><div id="body-link-popper-CR13" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">13. <span class="element-citation">Hobolth
 A. A Markov chain Monte Carlo expectation maximization algorithm for 
statistical analysis of DNA sequence evolution with neighbor-dependent 
substitution rates. <span><span class="ref-journal">J. Comput. Graph. Stat. </span>2008;<span class="ref-vol">17</span>(1):138–162. doi: 10.1198/106186008X289010.</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.1198%2F106186008X289010" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span> [<a href="#CR13">Ref list</a>]</div><div id="body-link-popper-CR14" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Identification and measurement of neighbor-dependent nucleotide substitution processes.<div class="authors">Arndt PF, Hwa T</div><div class="citation">Bioinformatics. 2005 May 15; 21(10):2322-8.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/15769841/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR14">Ref list</a>]</p></div><div id="body-link-popper-CR15" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Predicting evolutionary change in the influenza A virus.<div class="authors">Ferguson NM, Anderson RM</div><div class="citation">Nat Med. 2002 Jun; 8(6):562-3.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/12042802/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR15">Ref list</a>]</p></div><div id="body-link-popper-CR16" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Mapping the antigenic and genetic evolution of influenza virus.<div class="authors">Smith DJ, Lapedes AS, de Jong JC, Bestebroer TM, Rimmelzwaan GF, Osterhaus AD, Fouchier RA</div><div class="citation">Science. 2004 Jul 16; 305(5682):371-6.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/15218094/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR16">Ref list</a>]</p></div><div id="body-link-popper-CR17" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Molecular epidemiologic investigation of lentogenic Newcastle disease virus from domestic birds at live bird markets in Korea.<div class="authors">Choi KS, Lee EK, Jeon WJ, Kwon JH, Lee JH, Sung HW</div><div class="citation">Avian Dis. 2012 Mar; 56(1):218-23.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/22545549/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR17">Ref list</a>]</p></div><div id="body-link-popper-CR18" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Pathotypical
 characterization and molecular epidemiology of Newcastle disease virus 
isolates from different hosts in China from 1996 to 2005.<div class="authors">Qin ZM, Tan LT, Xu HY, Ma BC, Wang YL, Yuan XY, Liu WJ</div><div class="citation">J Clin Microbiol. 2008 Feb; 46(2):601-11.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/18077643/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR18">Ref list</a>]</p></div><div id="body-link-popper-CR19" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Predicting the functional effect of amino acid substitutions and indels.<div class="authors">Choi Y, Sims GE, Murphy S, Miller JR, Chan AP</div><div class="citation">PLoS One. 2012; 7(10):e46688.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/23056405/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR19">Ref list</a>]</p></div><div id="body-link-popper-CR20" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Enhanced prediction of lopinavir resistance from genotype by use of artificial neural networks.<div class="authors">Wang D, Larder B</div><div class="citation">J Infect Dis. 2003 Sep 1; 188(5):653-60.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/12934180/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR20">Ref list</a>]</p></div><div id="body-link-popper-CR21" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true"><span class="invert">Review</span> Computer prediction of drug resistance mutations in proteins.<div class="authors">Cao ZW, Han LY, Zheng CJ, Ji ZL, Chen X, Lin HH, Chen YZ</div><div class="citation">Drug Discov Today. 2005 Apr 1; 10(7):521-9.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/15809198/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR21">Ref list</a>]</p></div><div id="body-link-popper-CR22" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Diversity and complexity of HIV-1 drug resistance: a bioinformatics approach to predicting phenotype from genotype.<div class="authors">Beerenwinkel N, Schmidt B, Walter H, Kaiser R, Lengauer T, Hoffmann D, Korn K, Selbig J</div><div class="citation">Proc Natl Acad Sci U S A. 2002 Jun 11; 99(12):8271-6.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/12060770/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR22">Ref list</a>]</p></div><div id="body-link-popper-CR23" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Feature-based classifiers for somatic mutation detection in tumour-normal paired sequencing data.<div class="authors">Ding J, Bashashati A, Roth A, Oloumi A, Tse K, Zeng T, Haffari G, Hirst M, Marra MA, Condon A, Aparicio S, Shah SP</div><div class="citation">Bioinformatics. 2012 Jan 15; 28(2):167-75.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/22084253/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR23">Ref list</a>]</p></div><div id="body-link-popper-CR24" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true"><span class="invert">Review</span> On the importance of cotranscriptional RNA structure formation.<div class="authors">Lai D, Proctor JR, Meyer IM</div><div class="citation">RNA. 2013 Nov; 19(11):1461-73.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/24131802/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR24">Ref list</a>]</p></div><div id="body-link-popper-CR25" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true"><span class="invert">Review</span> Folding and finding RNA secondary structure.<div class="authors">Mathews DH, Moss WN, Turner DH</div><div class="citation">Cold Spring Harb Perspect Biol. 2010 Dec; 2(12):a003665.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/20685845/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR25">Ref list</a>]</p></div><div id="body-link-popper-CR26" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Secondary structure prediction for aligned RNA sequences.<div class="authors">Hofacker IL, Fekete M, Stadler PF</div><div class="citation">J Mol Biol. 2002 Jun 21; 319(5):1059-66.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/12079347/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR26">Ref list</a>]</p></div><div id="body-link-popper-CR27" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true"><span class="invert">Review</span> Mutational analysis in RNAs: comparing programs for RNA deleterious mutation prediction.<div class="authors">Barash D, Churkin A</div><div class="citation">Brief Bioinform. 2011 Mar; 12(2):104-14.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/21422070/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR27">Ref list</a>]</p></div><div id="body-link-popper-CR28" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Profiling the HeLa S3 transcriptome using randomly primed cDNA and massively parallel short-read sequencing.<div class="authors">Morin R, Bainbridge M, Fejes A, Hirst M, Krzywinski M, Pugh T, McDonald H, Varhol R, Jones S, Marra M</div><div class="citation">Biotechniques. 2008 Jul; 45(1):81-94.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/18611170/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR28">Ref list</a>]</p></div><div id="body-link-popper-CR29" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">SNVMix: predicting single nucleotide variants from next-generation sequencing of tumors.<div class="authors">Goya
 R, Sun MG, Morin RD, Leung G, Ha G, Wiegand KC, Senz J, Crisan A, Marra
 MA, Hirst M, Huntsman D, Murphy KP, Aparicio S, Shah SP</div><div class="citation">Bioinformatics. 2010 Mar 15; 26(6):730-6.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/20130035/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR29">Ref list</a>]</p></div><div id="body-link-popper-CR30" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Mapping short DNA sequencing reads and calling variants using mapping quality scores.<div class="authors">Li H, Ruan J, Durbin R</div><div class="citation">Genome Res. 2008 Nov; 18(11):1851-8.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/18714091/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR30">Ref list</a>]</p></div><div id="body-link-popper-CR31" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Accurate estimation of substitution rates with neighbor-dependent models in a phylogenetic context.<div class="authors">Bérard J, Guéguen L</div><div class="citation">Syst Biol. 2012 May; 61(3):510-21.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/22331438/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR31">Ref list</a>]</p></div><div id="body-link-popper-CR32" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Analysis of sequence and haemagglutinin activity of the HN glycoprotein of Newcastle disease virus.<div class="authors">Ke GM, Chuang KP, Chang CD, Lin MY, Liu HJ</div><div class="citation">Avian Pathol. 2010 Jun; 39(3):235-44.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/20544431/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR32">Ref list</a>]</p></div><div id="body-link-popper-CR33" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">33. <span class="element-citation">Bal M. Rough sets theory as symbolic data mining method: an application on complete decision table. <span><span class="ref-journal">Inform. Sci. Lett. </span>2013;<span class="ref-vol">2</span>(1):35–47. doi: 10.12785/isl/020105.</span> <span class="nowrap ref crossref">[<a href="https://dx.doi.org/10.12785%2Fisl%2F020105" target="pmc_ext" ref="reftype=other&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article%7CCitationRef&amp;TO=Content%20Provider%7CCrosslink%7CDOI&amp;rendering-type=normal">Cross Ref</a>]</span></span> [<a href="#CR33">Ref list</a>]</div><div id="body-link-popper-CR34" class="body-link-popper ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic" style="display: none; width: 30em; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true">Complete genome sequence of a newly emerging newcastle disease virus.<div class="authors">Wang JY, Liu WH, Ren JJ, Tang P, Wu N, Liu HJ</div><div class="citation">Genome Announc. 2013 May 9; 1(3):.</div><p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/23661479/" ref="reftype=pubmed&amp;article-id=4867776&amp;issue-id=285916&amp;journal-id=495&amp;FROM=Article|CitedRefBlock&amp;TO=Entrez|Pubmed|Record&amp;rendering-type=normal">PubMed</a>] [<a href="#CR34">Ref list</a>]</p></div></div>
                        </div>
                        <div class="bottom">
                            
                            <div id="NCBIFooter_dynamic">
    <a id="help-desk-link" class="help_desk jig-ncbihelpwindow" target="_blank" href="">Support Center</a>
    <a id="help-desk-link" class="help_desk jig-ncbihelpwindow" target="_blank" href="https://support.ncbi.nlm.nih.gov/ics/support/KBList.asp?Time=2017-06-22T23:32:26-04:00&amp;Snapshot=%2Fprojects%2FPMC%2FPMCViewer@4.45&amp;Host=ptpmc202&amp;ncbi_phid=F4FC0CDA94C8AC010000000000A2008E&amp;ncbi_session=3970AFB68BEF94A1_0040SID&amp;from=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2Fpmc%2Farticles%2FPMC4867776%2F&amp;Db=pmc&amp;folderID=132&amp;Ncbi_App=pmc&amp;Page=literature&amp;style=classic&amp;deptID=28049">Support Center</a>
    
</div>

                            <div class="footer" id="footer">
    
    <div class="subfooter"> </div><script type="text/javascript" src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/preloaderWidget.js"> </script>
    <div id="external-disclaimer" class="offscreen_noflow">
        External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.
    </div>    
    <div id="ncbifooter" class="contact_info">      
        <div id="footer-contents-right">
            <div id="nlm_thumb_logo">
                <a href="https://www.nlm.nih.gov/" title="NLM">NLM</a>
            </div>
            <div id="nih_thumb_logo">
                <a href="https://www.nih.gov/" title="NIH">NIH</a>
            </div>
            <div id="hhs_thumb_logo">
                <a href="https://www.hhs.gov/" title="DHHS">DHHS</a>
            </div>
            <div id="usagov_thumb_logo">
                <a href="https://www.usa.gov/" title="USA.gov">USA.gov</a>
            </div>         
        </div>
        
        <div id="footer-contents-left">
            <p class="address vcard">
                <span class="url">
                    <a class="fn url newdomain" href="https://www.ncbi.nlm.nih.gov/">National Center for
                        Biotechnology Information</a>,
                </span> <span class="org url newdomain"><a href="https://www.nlm.nih.gov/">U.S. National Library of Medicine</a></span>
                <span class="adr">
                    <span class="street-address">8600 Rockville Pike</span>, <span class="locality">Bethesda</span>
                    <span class="region">MD</span>, <span class="postal-code">20894</span>
                    <span class="country-name">USA</span>
                </span>
            </p>
            
            <a href="https://www.ncbi.nlm.nih.gov/home/about/policies.shtml">Policies and Guidelines</a> | <a href="https://www.ncbi.nlm.nih.gov/home/about/contact.shtml">Contact</a>
        </div>
    </div>
    <script type="text/javascript" src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/InstrumentPageStarterJS.js"> </script>    
    <script type="text/javascript" src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/hfjs2.js"> </script>
</div>
                        </div>
                    </div>
                    <!--/.page-->
                </div>
                <!--/.wrap-->
            </div><!-- /.twelve_col -->
        <div class="ui-ncbiautocomplete-holder shadow ui-ncbiautocomplete-holder-clearfix" aria-live="assertive" style="display: none; top: 65.6px; left: 370.7px; width: 670px; z-index: 1001;"><ul class="ui-ncbiautocomplete-options" role="listbox" aria-activedescendant="term"></ul><div class="ui-ncbiautocomplete-actions shadow" style="display: block;"><a class="ui-ncbiautocomplete-link-pref" style="display: none;">Preferences</a><a href="#" class="ui-ncbiautocomplete-link-off ui-ncbiautocomplete-link-pref-right" style="display: block;">Turn off</a></div></div><div style="display: none; top: -100px; left: -100px;" aria-live="assertive" aria-hidden="true" class="ui-helper-reset ui-ncbipopper-wrapper ui-ncbipopper-basic">External link. Please review our <a href="https://www.nlm.nih.gov/privacy.html">privacy policy</a>.</div></div><script async="1" src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/analytics.js"></script>
        <!-- /.grid -->

        <span class="PAFAppResources"></span>
        
        <!-- BESelector tab -->
        
        
        
        <noscript><img alt="statistics" 
src="/stat?jsdisabled=true&amp;ncbi_db=pmc&amp;ncbi_pdid=article&amp;ncbi_acc=&amp;ncbi_domain=ejbiosysbio&amp;ncbi_report=record&amp;ncbi_type=fulltext&amp;ncbi_objectid=&amp;ncbi_pcid=/articles/PMC4867776/&amp;ncbi_app=pmc"
 /></noscript>
        
        
        <!-- usually for JS scripts at page bottom -->
        <!--<component id="PageFixtures" label="styles"></component>-->
    

<!-- 3970AFB68BEF94A1_0040SID /projects/PMC/PMCViewer@4.45 ptpmc202 v4.1.r526566 Fri, Feb 03 2017 14:27:04 -->

<script type="text/javascript" src="The%20prediction%20of%20virus%20mutation%20using%20neural%20networks%20and%20rough%20set%20techniques_files/4065628.js" snapshot="pmc"></script>
<div class="ui-dialog ui-widget ui-widget-content ui-corner-all ui-front" style="display: none;" tabindex="-1" role="dialog" aria-describedby="epubDialog" aria-labelledby="ui-id-1"><div class="ui-dialog-titlebar ui-widget-header ui-corner-all ui-helper-clearfix"><span id="ui-id-1" class="ui-dialog-title">Making articles easier to read in PMC</span><button type="button" class="ui-button ui-widget ui-state-default ui-corner-all ui-button-icon-only ui-dialog-titlebar-close" role="button" title="Close"><span class="ui-button-icon-primary ui-icon ui-icon-closethick"></span><span class="ui-button-text">Close</span></button></div><div id="epubDialog" style="display: block;" class="ui-dialog-content ui-widget-content">  <p>We
 are experimenting with display styles that make it easier to read 
articles     in PMC.     Our first effort uses eBook readers, which have
 several "ease of reading"     features already built in.</p>  <p>These PMC articles are best viewed in the <em>iBooks     reader</em>. You may notice problems with the display of certain parts of an article     in other eReaders.</p>  <div class="ui-dialog-buttonpane ui-widget-content ui-helper-clearfix"><button id="cancelEpub" class="ui-state-default ui-corner-all">Cancel</button><button id="downloadEpub" style="float: left" class="ui-state-default ui-corner-all">Download article</button></div>          </div></div><ul class="print-log"><li></li></ul><div style="position: absolute; width: 0px; height: 0px; overflow: hidden; padding: 0px; border: 0px none; margin: 0px;"><div id="MathJax_Font_Test" style="position: absolute; visibility: hidden; top: 0px; left: 0px; width: auto; padding: 0px; border: 0px none; margin: 0px; white-space: nowrap; text-align: left; text-indent: 0px; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; font-size: 40px; font-weight: normal; font-style: normal; font-size-adjust: none; font-family: MathJax_Size2,sans-serif;"></div></div></body></html>